
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.4. HOMER FindPeaks &#8212; My sample book</title>
    
  <link rel="stylesheet" href="../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/togglebutton.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="2.5. HOMER FindcsRNATSS" href="04_homer_findTSS.html" />
    <link rel="prev" title="2.3. Create Genome Browser Visualization Files" href="03_homer_vis.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../index.html">
  
  <img src="../../_static/logo.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">My sample book</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../intro.html">
   Welcome to your Jupyter Book
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Example
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../content.html">
   Content in Jupyter Book
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../markdown.html">
     Markdown Files
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../notebooks.html">
     Content with notebooks
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Peak calling of PRO-cap data
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../11_prep_procap/01_prep_files.html">
   1. PRO-cap Data description and preparation
  </a>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="00_homer_intro.html">
   2. Run Homer Analysis on PRO-cap Data
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="01_prep_files.html">
     2.1. Check the file paths for the data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="02_homer_tag.html">
     2.2. Run HOMER makeTagDirectory of PRO-cap and RNA-seq data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="03_homer_vis.html">
     2.3. Create Genome Browser Visualization Files
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.4. HOMER FindPeaks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="04_homer_findTSS.html">
     2.5. HOMER FindcsRNATSS
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="05_homer_merge_peak.html">
     2.6. Merge Peak output from HOMER findPeaks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="05_homer_merge_tss.html">
     2.7. Merge TSS output from HOMER findcsRNATSS
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="06_homer_annot_peak.html">
     2.8. Get the count using HOMER annotatePeaks.pl
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="06_homer_annot_tss.html">
     2.9. Get the count using HOMER annotatePeaks.pl
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="12_peak_PER1.html">
     2.10. Summarize and visualize peaks from HOMER pipeline
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/notebooks/12_run_homer_procap/04_homer_findPeaks.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/executablebooks/jupyter-book"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fnotebooks/12_run_homer_procap/04_homer_findPeaks.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/docs/notebooks/12_run_homer_procap/04_homer_findPeaks.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#findpeaks-of-time-point-0-min">
   2.4.1. FindPeaks of time point 0 min
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-00-min-with-rna-control">
     2.4.1.1. Time point 00 min with RNA control
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-00-min-without-rna-control">
     2.4.1.2. Time point 00 min without RNA control
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#findpeaks-of-time-point-15-min">
   2.4.2. FindPeaks of time point 15 min
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-15-min-with-rna-control">
     2.4.2.1. Time point 15 min with RNA control
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-15-min-without-rna-control">
     2.4.2.2. Time point 15 min without RNA control
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#findpeaks-of-time-point-60-min">
   2.4.3. FindPeaks of time point 60 min
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-60-min-with-rna-control">
     2.4.3.1. Time point 60 min with RNA control
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#time-point-60-min-without-rna-control">
     2.4.3.2. Time point 60 min without RNA control
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#check-output-file">
   2.4.4. Check output file
  </a>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="homer-findpeaks">
<h1><span class="section-number">2.4. </span>HOMER FindPeaks<a class="headerlink" href="#homer-findpeaks" title="Permalink to this headline">¶</a></h1>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">General</span> <span class="n">analysis</span> <span class="n">options</span><span class="p">:</span>
	<span class="o">-</span><span class="n">o</span> <span class="o">&lt;</span><span class="n">filename</span><span class="o">|</span><span class="n">auto</span><span class="o">&gt;</span> <span class="p">(</span><span class="n">file</span> <span class="n">name</span> <span class="k">for</span> <span class="n">to</span> <span class="n">output</span> <span class="n">peaks</span><span class="p">,</span> <span class="n">default</span><span class="p">:</span> <span class="n">stdout</span><span class="p">)</span>
		<span class="s2">&quot;-o auto&quot;</span> <span class="n">will</span> <span class="n">send</span> <span class="n">output</span> <span class="n">to</span> <span class="s2">&quot;&lt;tag directory&gt;/peaks.txt&quot;</span><span class="p">,</span> <span class="s2">&quot;.../regions.txt&quot;</span><span class="p">,</span>
		<span class="ow">or</span> <span class="s2">&quot;.../transcripts.txt&quot;</span> <span class="n">depending</span> <span class="n">on</span> <span class="n">the</span> <span class="s2">&quot;-style&quot;</span> <span class="n">option</span>
	<span class="o">-</span><span class="n">style</span> <span class="o">&lt;</span><span class="n">option</span><span class="o">&gt;</span> <span class="p">(</span><span class="n">Specialized</span> <span class="n">options</span> <span class="k">for</span> <span class="n">specific</span> <span class="n">analysis</span> <span class="n">strategies</span><span class="p">)</span>
		<span class="n">factor</span> <span class="p">(</span><span class="n">transcription</span> <span class="n">factor</span> <span class="n">ChIP</span><span class="o">-</span><span class="n">Seq</span><span class="p">,</span> <span class="n">uses</span> <span class="o">-</span><span class="n">center</span><span class="p">,</span> <span class="n">output</span><span class="p">:</span> <span class="n">peaks</span><span class="o">.</span><span class="n">txt</span><span class="p">,</span>  <span class="n">default</span><span class="p">)</span>
		<span class="n">histone</span> <span class="p">(</span><span class="n">histone</span> <span class="n">modification</span> <span class="n">ChIP</span><span class="o">-</span><span class="n">Seq</span><span class="p">,</span> <span class="n">region</span> <span class="n">based</span><span class="p">,</span> <span class="n">uses</span> <span class="o">-</span><span class="n">region</span> <span class="o">-</span><span class="n">size</span> <span class="mi">500</span> <span class="o">-</span><span class="n">L</span> <span class="mi">0</span><span class="p">,</span> <span class="n">regions</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="n">groseq</span> <span class="p">(</span><span class="n">de</span> <span class="n">novo</span> <span class="n">transcript</span> <span class="n">identification</span> <span class="kn">from</span> <span class="nn">GroSeq</span> <span class="n">data</span><span class="p">,</span> <span class="n">transcripts</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
	<span class="n">tss</span> <span class="p">(</span><span class="n">TSS</span> <span class="n">identification</span> <span class="kn">from</span> <span class="mi">5</span><span class="s1">&#39; RNA sequencing, tss.txt)</span>
		<span class="n">dnase</span> <span class="p">(</span><span class="n">Hypersensitivity</span> <span class="p">[</span><span class="n">crawford</span> <span class="n">style</span> <span class="p">(</span><span class="n">nicking</span><span class="p">)],</span> <span class="n">peaks</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="nb">super</span> <span class="p">(</span><span class="n">Super</span> <span class="n">Enhancers</span><span class="p">,</span> <span class="n">superEnhancers</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="n">superhistone</span> <span class="p">(</span><span class="n">Super</span> <span class="n">Enhancers</span> <span class="kn">from</span> <span class="nn">H3K27ac</span> <span class="n">data</span><span class="p">,</span> <span class="n">superEnhancers</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="n">mC</span> <span class="p">(</span><span class="n">Cytosine</span> <span class="n">methylation</span> <span class="p">(</span><span class="n">BS</span><span class="o">-</span><span class="n">seq</span><span class="o">/</span><span class="n">methylC</span><span class="o">-</span><span class="n">seq</span><span class="p">),</span> <span class="n">regions</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="n">damid</span> <span class="p">(</span><span class="n">DamID</span> <span class="n">enrichment</span> <span class="kn">from</span> <span class="nn">DpnI</span> <span class="n">digestion</span><span class="p">,</span> <span class="n">regions</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
		<span class="n">clip</span> <span class="p">(</span><span class="n">CLIP</span><span class="o">-</span><span class="n">Seq</span> <span class="n">enrichment</span><span class="p">,</span> <span class="n">strand</span> <span class="n">specific</span><span class="p">,</span> <span class="n">peaks</span><span class="o">.</span><span class="n">txt</span><span class="p">)</span>
</pre></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="nb">print</span><span class="p">(</span><span class="mi">1</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>1
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
findPeaks 
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Usage: findPeaks &lt;tag directory&gt; [options]

	Finds peaks in the provided tag directory.  By default, peak list printed to stdout

	General analysis options:
		-o &lt;filename|auto&gt; (file name for to output peaks, default: stdout)
			&quot;-o auto&quot; will send output to &quot;&lt;tag directory&gt;/peaks.txt&quot;, &quot;.../regions.txt&quot;,
			or &quot;.../transcripts.txt&quot; depending on the &quot;-style&quot; option
		-style &lt;option&gt; (Specialized options for specific analysis strategies)
			factor (transcription factor ChIP-Seq, uses -center, output: peaks.txt,  default)
			histone (histone modification ChIP-Seq, region based, uses -region -size 500 -L 0, regions.txt)
			groseq (de novo transcript identification from GroSeq data, transcripts.txt)
			tss (TSS identification from 5&#39; RNA sequencing, tss.txt)
			dnase (Hypersensitivity [crawford style (nicking)], peaks.txt)
			super (Super Enhancers, superEnhancers.txt)
			superhistone (Super Enhancers from H3K27ac data, superEnhancers.txt)
			mC (Cytosine methylation (BS-seq/methylC-seq), regions.txt)
			damid (DamID enrichment from DpnI digestion, regions.txt)
			clip (CLIP-Seq enrichment, strand specific, peaks.txt)

	chipseq/histone options:
		-i &lt;input tag directory&gt; (Experiment to use as IgG/Input/Control)
		-size &lt;#&gt; (Peak size, default: auto)
		-minDist &lt;#&gt; (minimum distance between peaks, default: peak size x2)
		-gsize &lt;#&gt; (Set effective mappable genome size, default: 2e9)
		-fragLength &lt;#|auto&gt; (Approximate fragment length, default: auto)
		-inputFragLength &lt;#|auto&gt; (Approximate fragment length of input tags, default: auto)
		-tbp &lt;#&gt; (Maximum tags per bp to count, 0 = no limit, default: auto)
		-inputtbp &lt;#&gt; (Maximum tags per bp to count in input, 0 = no limit, default: auto)
		-strand &lt;both|separate&gt; (find peaks using tags on both strands or separate, default:both)
		-norm # (Tag count to normalize to, default 10000000)
		-region (extends start/stop coordinates to cover full region considered &quot;enriched&quot;)
			-regionRes &lt;#&gt; (number of fractions peaks are divided in when extending &#39;regions&#39;, def: 4)
		-center (Centers peaks on maximum tag overlap and calculates focus ratios)
		-nfr (Centers peaks on most likely nucleosome free region [works best with mnase data])
			(-center and -nfr can be performed later with &quot;getPeakTags&quot;

	Peak Filtering options: (set -F/-L/-C to 0 to skip)
		-F &lt;#&gt; (fold enrichment over input tag count, default: 4.0)
		  -P &lt;#&gt; (poisson p-value threshold relative to input tag count, default: 0.0001)
		-L &lt;#&gt; (fold enrichment over local tag count, default: 4.0)
		  -LP &lt;#&gt; (poisson p-value threshold relative to local tag count, default: 0.0001)
		-C &lt;#&gt; (fold enrichment limit of expected unique tag positions, default: 2.0)
		-localSize &lt;#&gt; (region to check for local tag enrichment, default: 10000)
		-inputSize &lt;#&gt; (Size of region to search for control tags, default: 2x peak size)
		-fdr &lt;#&gt; (False discovery rate, default = 0.001)
		-poisson &lt;#&gt; (Set poisson p-value cutoff, default: uses fdr)
		-tagThreshold &lt;#&gt; (Set # of tags to define a peak, default: 25)
		-ntagThreshold &lt;#&gt; (Set # of normalized tags to define a peak, by default uses 1e7 for norm)
		-minTagThreshold &lt;#&gt; (Absolute minimum tags per peak, default: expected tags per peak)

	SuperEnhancer Options: (Need to specify &quot;-style super&quot;):
		-superSlope &lt;#&gt; (Slope threshold to identify super vs. typical enh., default: 1.00)
		-superWindow &lt;#&gt; (moving window/number of peaks to use to calculate slope, default: 10)
		-typical &lt;filename&gt; (Output typical enhancers to this file, default: not used)
		-inputPeaks &lt;filename&gt; (initial peaks to use for super enhancer merging/scoring)
		-excludePeaks &lt;filename&gt; (regions to exclude from analysis, i.e. TSS regions for H3K27ac)

	MethylC-Seq/BS-Seq options (Need to specify &quot;-style mC&quot;):
		-unmethylC / -methylC (find unmethylated/methylated regions, default: -unmethyC)
		-mCthresh &lt;#&gt; (methylation threshold of regions, default: avg methylation/2)
		-minNumC &lt;#&gt; (Minimum number of cytosines per methylation peak, default: 6)

	GroSeq Options (Need to specify &quot;-style groseq&quot;):
		-tssSize &lt;#&gt; (size of region for initiation detection/artifact size, default: 250)
		-minBodySize &lt;#&gt; (size of regoin for transcript body detection, default: 1000)
		-tssFold &lt;#&gt; (fold enrichment for new initiation dectection, default: 4.0)
		-bodyFold &lt;#&gt; (fold enrichment for new transcript dectection, default: 4.0)
		-endFold &lt;#&gt; (end transcript when levels are this much less than the start, default: 10.0)
		-method &lt;fold|level&gt; (method used for identifying new transcripts, default: fold)
		-fragLength &lt;#&gt; (Approximate fragment length, default: 150)
		-uniqmap &lt;directory&gt; (directory of binary files specifying uniquely mappable locations)
			Download from http://biowhat.ucsd.edu/homer/groseq/
		-confPvalue &lt;#&gt; (confidence p-value: 1.00e-05)
		-minReadDepth &lt;#&gt; (Minimum initial read depth for transcripts, default: auto)
		-pseudoCount &lt;#&gt; (Pseudo tag count, default: 2.0)
		-rev (reverse strand of reads - for first-strand rna-seq/gro-seq)
		-gtf &lt;filename&gt; (Output de novo transcripts in GTF format)
			&quot;-o auto&quot; will produce &lt;dir&gt;/transcripts.txt and &lt;dir&gt;/transcripts.gtf
		
</pre></div>
</div>
</div>
</div>
<div class="section" id="findpeaks-of-time-point-0-min">
<h2><span class="section-number">2.4.1. </span>FindPeaks of time point 0 min<a class="headerlink" href="#findpeaks-of-time-point-0-min" title="Permalink to this headline">¶</a></h2>
<div class="section" id="time-point-00-min-with-rna-control">
<h3><span class="section-number">2.4.1.1. </span>Time point 00 min with RNA control<a class="headerlink" href="#time-point-00-min-with-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap00_rna00.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap00_rna00.txt
findPeaks $FD_TAG_CAP_T00/ -i $FD_TAG_RNA_T00/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751135
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="time-point-00-min-without-rna-control">
<h3><span class="section-number">2.4.1.2. </span>Time point 00 min without RNA control<a class="headerlink" href="#time-point-00-min-without-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap00.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap00.txt
findPeaks $FD_TAG_CAP_T00/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751136
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="findpeaks-of-time-point-15-min">
<h2><span class="section-number">2.4.2. </span>FindPeaks of time point 15 min<a class="headerlink" href="#findpeaks-of-time-point-15-min" title="Permalink to this headline">¶</a></h2>
<div class="section" id="time-point-15-min-with-rna-control">
<h3><span class="section-number">2.4.2.1. </span>Time point 15 min with RNA control<a class="headerlink" href="#time-point-15-min-with-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap15_rna15.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap15_rna15.txt
findPeaks $FD_TAG_CAP_T15/ -i $FD_TAG_RNA_T15/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751137
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="time-point-15-min-without-rna-control">
<h3><span class="section-number">2.4.2.2. </span>Time point 15 min without RNA control<a class="headerlink" href="#time-point-15-min-without-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap15.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap15.txt
findPeaks $FD_TAG_CAP_T15/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751138
</pre></div>
</div>
</div>
</div>
</div>
</div>
<div class="section" id="findpeaks-of-time-point-60-min">
<h2><span class="section-number">2.4.3. </span>FindPeaks of time point 60 min<a class="headerlink" href="#findpeaks-of-time-point-60-min" title="Permalink to this headline">¶</a></h2>
<div class="section" id="time-point-60-min-with-rna-control">
<h3><span class="section-number">2.4.3.1. </span>Time point 60 min with RNA control<a class="headerlink" href="#time-point-60-min-with-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap60_rna60.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap60_rna60.txt
findPeaks $FD_TAG_CAP_T60/ -i $FD_TAG_RNA_T60/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751139
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="time-point-60-min-without-rna-control">
<h3><span class="section-number">2.4.3.2. </span>Time point 60 min without RNA control<a class="headerlink" href="#time-point-60-min-without-rna-control" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findPeaks_cap60.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### Set Directories
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer
FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60
FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15
FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00
FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60
FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15
FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00

#findPeaks &lt;tag directory&gt; -style &lt;factor | histone&gt; -o auto -i &lt;control tag directory&gt;
#findPeaks Exp1-csRNA-TagDir/ -i Exp1-input-TagDir/ -style tss &gt; tssOutput.txt
#findPeaks IMR90-5GROseq/ -o auto -style tss -i IMR90-GROseq/
FP_OUT=$FD_OUT/out_findPeak/tss_cap60.txt
findPeaks $FD_TAG_CAP_T60/ -style tss &gt; $FP_OUT

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24751140
</pre></div>
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="check-output-file">
<h2><span class="section-number">2.4.4. </span>Check output file<a class="headerlink" href="#check-output-file" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head -40 /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findPeak/tss_cap00_rna00.txt
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># HOMER Peaks
# Peak finding parameters:
# tag directory = /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/
#
# total peaks = 55894
# peak size = 150
# peaks found separately for each strand
# minimum distance between peaks = 300
# fragment length = 0
# genome size = 2000000000
# Total tags = 9384553.0
# Total tags in peaks = 4447460.5
# Approximate IP efficiency = 47.39%
# tags per bp = 0.004692
# expected tags per peak = 0.352
# maximum tags considered per bp = 0.0
# effective number of tags used for normalization = 10000000.0
# Peaks have been centered at maximum tag pile-up
# FDR rate threshold = 0.001000000
# FDR effective poisson threshold = 9.756258e-08
# FDR tag threshold = 7.0
# number of putative peaks = 70681
#
# input tag directory = /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/rnaseq_t00/
# Fold over input required = 4.00
# Poisson p-value over input required = 1.00e-04
# Putative peaks filtered by input = 5853
#
# size of region used for local filtering = 10000
# Fold over local region required = 4.00
# Poisson p-value over local region required = 1.00e-04
# Putative peaks filtered by local signal = 8934
#
# cmd = findPeaks /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/ -i /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/rnaseq_t00/ -style tss
#
# Column Headers:
#PeakID	chr	start	end	strand	Normalized Tag Count	focus ratio	findPeaks Score	Total Tags	Control Tags (normalized to IP Experiment)	Fold Change vs Control	p-value vs Control	Fold Change vs Local	p-value vs Local	Dispersion Ratio	Periodic Ratio
chr14-1	chr14	49862475	49862625	+	6671.1	0.974	6394.000000	6260.5	8.7	721.73	0.00e+00	924.87	0.00e+00	0.970764	0.513661
chr14-2	chr14	49862774	49862924	-	6293.9	0.314	6046.000000	5906.5	6.5	906.17	0.00e+00	487.87	0.00e+00	0.368860	0.391908
chr14-3	chr14	49586803	49586953	-	4523.4	0.887	4290.500000	4245.0	10.3	413.72	0.00e+00	640.82	0.00e+00	0.894532	0.446809
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head -40 /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findPeak/tss_cap00.txt
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span># HOMER Peaks
# Peak finding parameters:
# tag directory = /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/
#
# total peaks = 60080
# peak size = 150
# peaks found separately for each strand
# minimum distance between peaks = 300
# fragment length = 0
# genome size = 2000000000
# Total tags = 9384553.0
# Total tags in peaks = 4845493.5
# Approximate IP efficiency = 51.63%
# tags per bp = 0.004692
# expected tags per peak = 0.352
# maximum tags considered per bp = 0.0
# effective number of tags used for normalization = 10000000.0
# Peaks have been centered at maximum tag pile-up
# FDR rate threshold = 0.001000000
# FDR effective poisson threshold = 9.756258e-08
# FDR tag threshold = 7.0
# number of putative peaks = 70681
#
# size of region used for local filtering = 10000
# Fold over local region required = 4.00
# Poisson p-value over local region required = 1.00e-04
# Putative peaks filtered by local signal = 10601
#
# cmd = findPeaks /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/ -style tss
#
# Column Headers:
#PeakID	chr	start	end	strand	Normalized Tag Count	focus ratio	findPeaks Score	Fold Change vs Local	p-value vs Local	Dispersion Ratio	Periodic Ratio
chr14-1	chr14	49862475	49862625	+	6671.1	0.974	6394.000000	924.87	0.00e+00	0.970764	0.513661
chr14-2	chr14	49862774	49862924	-	6293.9	0.314	6046.000000	487.87	0.00e+00	0.368860	0.391908
chr14-3	chr14	49586803	49586953	-	4523.4	0.887	4290.500000	640.82	0.00e+00	0.894532	0.446809
chr17-1	chr17	19187935	19188085	+	4359.3	0.945	4116.000000	1859.12	0.00e+00	0.928519	0.379074
chr14-4	chr14	49586499	49586649	+	4258.1	0.797	4075.000000	383.07	0.00e+00	0.769666	0.488816
chr19-1	chr19	33302460	33302610	-	4256.5	0.899	3995.500000	1686.85	0.00e+00	0.826930	0.512654
chr19-2	chr19	33302394	33302544	+	4151.5	0.845	3939.000000	1005.25	0.00e+00	0.536833	0.502078
chr17-2	chr17	19188157	19188307	-	3861.7	0.240	3668.000000	388.85	0.00e+00	0.259299	0.590101
</pre></div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span>  <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cname</span> <span class="o">=</span> <span class="s2">&quot;PeakID	chr	start	end	strand	Normalized Tag Count	focus ratio	findPeaks Score	Total Tags	Control Tags (normalized to IP Experiment)	Fold Change vs Control	p-value vs Control	Fold Change vs Local	p-value vs Local	Dispersion Ratio	Periodic Ratio&quot;</span>
<span class="n">cname</span> <span class="o">=</span> <span class="n">cname</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fpath</span> <span class="o">=</span> <span class="s2">&quot;/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findPeak/tss_t00.txt&quot;</span>
<span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">,</span> <span class="n">comment</span><span class="o">=</span><span class="s2">&quot;#&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PeakID</th>
      <th>chr</th>
      <th>start</th>
      <th>end</th>
      <th>strand</th>
      <th>Normalized Tag Count</th>
      <th>focus ratio</th>
      <th>findPeaks Score</th>
      <th>Total Tags</th>
      <th>Control Tags (normalized to IP Experiment)</th>
      <th>Fold Change vs Control</th>
      <th>p-value vs Control</th>
      <th>Fold Change vs Local</th>
      <th>p-value vs Local</th>
      <th>Dispersion Ratio</th>
      <th>Periodic Ratio</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>chr14-1</td>
      <td>chr14</td>
      <td>49862475</td>
      <td>49862625</td>
      <td>+</td>
      <td>6671.1</td>
      <td>0.974</td>
      <td>6394.0</td>
      <td>6260.5</td>
      <td>8.7</td>
      <td>721.73</td>
      <td>0.0</td>
      <td>924.87</td>
      <td>0.0</td>
      <td>0.970764</td>
      <td>0.513661</td>
    </tr>
    <tr>
      <th>1</th>
      <td>chr14-2</td>
      <td>chr14</td>
      <td>49862774</td>
      <td>49862924</td>
      <td>-</td>
      <td>6293.9</td>
      <td>0.314</td>
      <td>6046.0</td>
      <td>5906.5</td>
      <td>6.5</td>
      <td>906.17</td>
      <td>0.0</td>
      <td>487.87</td>
      <td>0.0</td>
      <td>0.368860</td>
      <td>0.391908</td>
    </tr>
    <tr>
      <th>2</th>
      <td>chr14-3</td>
      <td>chr14</td>
      <td>49586803</td>
      <td>49586953</td>
      <td>-</td>
      <td>4523.4</td>
      <td>0.887</td>
      <td>4290.5</td>
      <td>4245.0</td>
      <td>10.3</td>
      <td>413.72</td>
      <td>0.0</td>
      <td>640.82</td>
      <td>0.0</td>
      <td>0.894532</td>
      <td>0.446809</td>
    </tr>
    <tr>
      <th>3</th>
      <td>chr17-1</td>
      <td>chr17</td>
      <td>19187935</td>
      <td>19188085</td>
      <td>+</td>
      <td>4359.3</td>
      <td>0.945</td>
      <td>4116.0</td>
      <td>4091.0</td>
      <td>0.0</td>
      <td>165068.08</td>
      <td>0.0</td>
      <td>1859.12</td>
      <td>0.0</td>
      <td>0.928519</td>
      <td>0.379074</td>
    </tr>
    <tr>
      <th>4</th>
      <td>chr14-4</td>
      <td>chr14</td>
      <td>49586499</td>
      <td>49586649</td>
      <td>+</td>
      <td>4258.1</td>
      <td>0.797</td>
      <td>4075.0</td>
      <td>3996.0</td>
      <td>9.6</td>
      <td>415.55</td>
      <td>0.0</td>
      <td>383.07</td>
      <td>0.0</td>
      <td>0.769666</td>
      <td>0.488816</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="p">[</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;chr&#39;</span><span class="p">]</span> <span class="o">==</span> <span class="s2">&quot;chr1&quot;</span><span class="p">]</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>PeakID</th>
      <th>chr</th>
      <th>start</th>
      <th>end</th>
      <th>strand</th>
      <th>Normalized Tag Count</th>
      <th>focus ratio</th>
      <th>findPeaks Score</th>
      <th>Total Tags</th>
      <th>Control Tags (normalized to IP Experiment)</th>
      <th>Fold Change vs Control</th>
      <th>p-value vs Control</th>
      <th>Fold Change vs Local</th>
      <th>p-value vs Local</th>
      <th>Dispersion Ratio</th>
      <th>Periodic Ratio</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>20</th>
      <td>chr1-1</td>
      <td>chr1</td>
      <td>8026234</td>
      <td>8026384</td>
      <td>-</td>
      <td>2015.00</td>
      <td>0.911</td>
      <td>1909.0</td>
      <td>1891.0</td>
      <td>23.4</td>
      <td>80.91</td>
      <td>0.000000e+00</td>
      <td>398.64</td>
      <td>0.000000e+00</td>
      <td>0.834658</td>
      <td>0.536859</td>
    </tr>
    <tr>
      <th>21</th>
      <td>chr1-2</td>
      <td>chr1</td>
      <td>8026190</td>
      <td>8026340</td>
      <td>+</td>
      <td>1973.50</td>
      <td>0.867</td>
      <td>1888.5</td>
      <td>1852.0</td>
      <td>1.9</td>
      <td>958.03</td>
      <td>0.000000e+00</td>
      <td>346.97</td>
      <td>0.000000e+00</td>
      <td>0.323072</td>
      <td>0.457452</td>
    </tr>
    <tr>
      <th>32</th>
      <td>chr1-3</td>
      <td>chr1</td>
      <td>149886837</td>
      <td>149886987</td>
      <td>+</td>
      <td>1644.70</td>
      <td>0.899</td>
      <td>1575.0</td>
      <td>1543.5</td>
      <td>2.5</td>
      <td>629.08</td>
      <td>0.000000e+00</td>
      <td>39.46</td>
      <td>0.000000e+00</td>
      <td>0.884279</td>
      <td>0.249300</td>
    </tr>
    <tr>
      <th>46</th>
      <td>chr1-5</td>
      <td>chr1</td>
      <td>28505962</td>
      <td>28506112</td>
      <td>+</td>
      <td>1447.10</td>
      <td>0.986</td>
      <td>1359.0</td>
      <td>1358.0</td>
      <td>98.3</td>
      <td>13.82</td>
      <td>0.000000e+00</td>
      <td>161.55</td>
      <td>0.000000e+00</td>
      <td>0.962077</td>
      <td>0.485437</td>
    </tr>
    <tr>
      <th>49</th>
      <td>chr1-6</td>
      <td>chr1</td>
      <td>153990729</td>
      <td>153990879</td>
      <td>-</td>
      <td>1382.10</td>
      <td>0.781</td>
      <td>1355.5</td>
      <td>1297.0</td>
      <td>1.1</td>
      <td>1189.38</td>
      <td>0.000000e+00</td>
      <td>215.35</td>
      <td>0.000000e+00</td>
      <td>0.238794</td>
      <td>0.534010</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>55866</th>
      <td>chr1-6619</td>
      <td>chr1</td>
      <td>94744028</td>
      <td>94744178</td>
      <td>-</td>
      <td>5.33</td>
      <td>0.524</td>
      <td>7.5</td>
      <td>5.0</td>
      <td>0.0</td>
      <td>201.75</td>
      <td>7.600000e-11</td>
      <td>54.72</td>
      <td>4.840000e-08</td>
      <td>0.777778</td>
      <td>1.000000</td>
    </tr>
    <tr>
      <th>55867</th>
      <td>chr1-6931</td>
      <td>chr1</td>
      <td>147305158</td>
      <td>147305308</td>
      <td>-</td>
      <td>5.33</td>
      <td>0.385</td>
      <td>7.0</td>
      <td>5.0</td>
      <td>0.0</td>
      <td>201.75</td>
      <td>7.600000e-11</td>
      <td>17.75</td>
      <td>1.120000e-05</td>
      <td>0.333333</td>
      <td>0.500000</td>
    </tr>
    <tr>
      <th>55868</th>
      <td>chr1-6998</td>
      <td>chr1</td>
      <td>185892856</td>
      <td>185893006</td>
      <td>-</td>
      <td>5.33</td>
      <td>0.542</td>
      <td>7.0</td>
      <td>5.0</td>
      <td>0.0</td>
      <td>201.75</td>
      <td>7.600000e-11</td>
      <td>26.27</td>
      <td>1.720000e-06</td>
      <td>0.100000</td>
      <td>0.555556</td>
    </tr>
    <tr>
      <th>55869</th>
      <td>chr1-7043</td>
      <td>chr1</td>
      <td>229093767</td>
      <td>229093917</td>
      <td>-</td>
      <td>5.33</td>
      <td>0.826</td>
      <td>7.0</td>
      <td>5.0</td>
      <td>0.0</td>
      <td>201.75</td>
      <td>7.600000e-11</td>
      <td>54.72</td>
      <td>4.840000e-08</td>
      <td>0.363636</td>
      <td>0.571429</td>
    </tr>
    <tr>
      <th>55870</th>
      <td>chr1-7044</td>
      <td>chr1</td>
      <td>229179115</td>
      <td>229179265</td>
      <td>+</td>
      <td>5.33</td>
      <td>0.632</td>
      <td>7.0</td>
      <td>5.0</td>
      <td>0.0</td>
      <td>201.75</td>
      <td>7.600000e-11</td>
      <td>32.83</td>
      <td>5.860000e-07</td>
      <td>0.444444</td>
      <td>0.600000</td>
    </tr>
  </tbody>
</table>
<p>5545 rows × 16 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;Control Tags (normalized to IP Experiment)&quot;</span><span class="p">]</span><span class="o">.</span><span class="n">hist</span><span class="p">(</span><span class="n">cumulative</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">density</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;AxesSubplot:&gt;
</pre></div>
</div>
<img alt="../../_images/04_homer_findPeaks_28_1.png" src="../../_images/04_homer_findPeaks_28_1.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">np</span><span class="o">.</span><span class="n">log2</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s2">&quot;Control Tags (normalized to IP Experiment)&quot;</span><span class="p">]</span> <span class="o">+</span> <span class="mi">1</span><span class="p">)</span><span class="o">.</span><span class="n">hist</span><span class="p">(</span><span class="n">cumulative</span><span class="o">=</span><span class="kc">False</span><span class="p">,</span> <span class="n">density</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">bins</span><span class="o">=</span><span class="mi">100</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;AxesSubplot:&gt;
</pre></div>
</div>
<img alt="../../_images/04_homer_findPeaks_29_1.png" src="../../_images/04_homer_findPeaks_29_1.png" />
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./notebooks/12_run_homer_procap"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="03_homer_vis.html" title="previous page"><span class="section-number">2.3. </span>Create Genome Browser Visualization Files</a>
    <a class='right-next' id="next-link" href="04_homer_findTSS.html" title="next page"><span class="section-number">2.5. </span>HOMER FindcsRNATSS</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The Jupyter Book Community<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>