{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HOMER FindcsRNATSS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "```\n",
    "\n",
    "\tfindcsRNATSS.pl <csRNA tag directory> [options]\n",
    "\n",
    "\tOptions:\n",
    "\t\t-o <prefix> \n",
    "\t\t-i <csRNA input tag directory>\n",
    "\t\t-rna <RNAseq tag directory>\n",
    "\t\t-gtf <gtf file>\n",
    "\t\t-genome <genome>\n",
    "\t\t-cpu <#> (max CPUs)\n",
    "\t\t-minDistDiff <#> (default: 0.15)\n",
    "\t\t-defaultLog2Fold <#> (default: 1)\n",
    "\t\t-maxInputLog2Fold <#> (maximum log2 fold enrichment vs. input or RNA, default: 3)\n",
    "\t\t-maxRNALog2Fold <#> (maximum log2 fold enrichment vs. input or RNA, default: 3)\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1\n"
     ]
    }
   ],
   "source": [
    "print(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tfindcsRNATSS.pl <csRNA tag directory> [options]\n",
      "\n",
      "\tOptions:\n",
      "\t\t-o <prefix> \n",
      "\t\t-i <csRNA input tag directory>\n",
      "\t\t-rna <RNAseq tag directory>\n",
      "\t\t-gtf <gtf file>\n",
      "\t\t-genome <genome>\n",
      "\t\t-cpu <#> (max CPUs)\n",
      "\t\t-minDistDiff <#> (default: 0.15)\n",
      "\t\t-defaultLog2Fold <#> (default: 1)\n",
      "\t\t-maxInputLog2Fold <#> (maximum log2 fold enrichment vs. input or RNA, default: 3)\n",
      "\t\t-maxRNALog2Fold <#> (maximum log2 fold enrichment vs. input or RNA, default: 3)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "findcsRNATSS.pl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "test if I did not provide input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FindcsRNATSS of time point 0 min"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 00 min with RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852983\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap00_rna00.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap00_rna00/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T00/ \\\n",
    "        -rna      $FD_TAG_RNA_T00/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 00 min without RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852984\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap00.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap00/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T00/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FindcsRNATSS of time point 15 min"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 15 min with RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852985\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap15_rna15.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap15_rna15/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T15/ \\\n",
    "        -rna      $FD_TAG_RNA_T15/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 15 min without RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852986\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap15.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap15/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T15/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## FindcsRNATSS of time point 60 min"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 60 min with RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852987\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60_rna60.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap60_rna60/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T60/ \\\n",
    "        -rna      $FD_TAG_RNA_T60/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Time point 60 min without RNA control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 24852988\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "module load perl\n",
    "module load gcc\n",
    "source /data/reddylab/software/miniconda2/bin/activate alex_dev\n",
    "export PATH=/data/reddylab/software/homer/bin/:$PATH\n",
    "sbatch -pnew,all \\\n",
    "    --mem 24G \\\n",
    "    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60.txt \\\n",
    "    <<'EOF'\n",
    "#!/bin/bash\n",
    "\n",
    "### Set Directories\n",
    "FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer\n",
    "FD_TAG_CAP_T60=$FD_OUT/tags/procap_t60\n",
    "FD_TAG_CAP_T15=$FD_OUT/tags/procap_t15\n",
    "FD_TAG_CAP_T00=$FD_OUT/tags/procap_t00\n",
    "FD_TAG_RNA_T60=$FD_OUT/tags/rnaseq_t60\n",
    "FD_TAG_RNA_T15=$FD_OUT/tags/rnaseq_t15\n",
    "FD_TAG_RNA_T00=$FD_OUT/tags/rnaseq_t00\n",
    "\n",
    "\n",
    "#findcsRNATSS.pl Exp1-csRNA-TagDir/ \n",
    "#    -o      outputPrefix \n",
    "#    -i      Exp1-input-TagDir/ \n",
    "#    -rna    Exp1-totalRNAseq-TagDir \n",
    "#    -gtf    genes.gtf \n",
    "#    -genome genome.fasta\n",
    "PREFIX=$FD_OUT/out_findTSS/tss_cap60/out\n",
    "\n",
    "findcsRNATSS.pl   $FD_TAG_CAP_T60/ \\\n",
    "        -o        $PREFIX \\\n",
    "        -genome   hg38\n",
    "\n",
    "EOF"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-----"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Check the results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ls: cannot access /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap*.txt: No such file or directory\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for FP in $(ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap*.txt); do\n",
    "    ls $FP\n",
    "    #cat $FP | grep Difference\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "head: cannot open `/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60.txt' for reading: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!head /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tail: cannot open `/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60.txt' for reading: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "!tail /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/findcsRNATSS_cap60.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap*: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "ls -d /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap00/*tsv: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap00/*tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap00/*txt: No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_cap00/*txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ls: cannot access /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt: No such file or directory\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt\\n'' returned non-zero exit status 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-15-224049e840f3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'bash'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt\\n'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/data/reddylab/Kuei/envs/jupyterhub/lib/python3.6/site-packages/IPython/core/interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[0;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[1;32m   2369\u001b[0m             \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2371\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2372\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2373\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/reddylab/Kuei/envs/jupyterhub/lib/python3.6/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[0;34m(line, cell)\u001b[0m\n\u001b[1;32m    140\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 142\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    143\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    144\u001b[0m         \u001b[0;31m# write a basic docstring:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<decorator-gen-110>\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n",
      "\u001b[0;32m/data/reddylab/Kuei/envs/jupyterhub/lib/python3.6/site-packages/IPython/core/magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(f, *a, **k)\u001b[0m\n\u001b[1;32m    185\u001b[0m     \u001b[0;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    186\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 187\u001b[0;31m         \u001b[0mcall\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0ma\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    188\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    189\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/data/reddylab/Kuei/envs/jupyterhub/lib/python3.6/site-packages/IPython/core/magics/script.py\u001b[0m in \u001b[0;36mshebang\u001b[0;34m(self, line, cell)\u001b[0m\n\u001b[1;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    244\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m!=\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 245\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0merr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    246\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    247\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCalledProcessError\u001b[0m: Command 'b'ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt\\n'' returned non-zero exit status 2."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "ls /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00_maxlen/out.inputDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00/out.inputDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00_maxlen/out.inputDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.inputDistribution.txt <==\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.inputDistribution.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert-danger\">\n",
    "The problem I encounter here is that all out.inputDistribution.txt is empty. My guess: the reason might be: `Difference in distributions is too small`\n",
    "</div>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00_maxlen/out.rnaDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00/out.rnaDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00_maxlen/out.rnaDistribution.txt <==\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.rnaDistribution.txt <==\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.rnaDistribution.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00_maxlen/out.stats.txt <==\n",
      "Set\tTotal TSS\n",
      "total\t52038\n",
      "valid TSS\t1074\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00/out.stats.txt <==\n",
      "Set\tTotal TSS\n",
      "total\t49006\n",
      "valid TSS\t594\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00_maxlen/out.stats.txt <==\n",
      "Set\tTotal TSS\n",
      "total\t58065\n",
      "valid TSS\t26732\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.stats.txt <==\n",
      "Set\tTotal TSS\n",
      "total\t55174\n",
      "valid TSS\t23771\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head -3 /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.stats.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Set\tTotal TSS\n",
      "total\t52038\n",
      "valid TSS\t1074\n",
      "total over input\t1088\n",
      " under RNA\t14\n",
      " valid Annotation: 1088\n",
      " exon Annotation: 0\n",
      "Total input\t50950\n",
      "log2 fold vs. input: 1\n",
      "log2 fold vs. rna: 1\n",
      "Number of TSS regions for foldChange Calculation: 0\n",
      "Number of exon regions for foldChange Calculation: 0\n"
     ]
    }
   ],
   "source": [
    "cat /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00_maxlen/out.stats.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00_maxlen/out.tss.txt <==\n",
      "#tssID\tchr\tstart\tend\tstrand\tscore\tfocusRatio\tcsRNA\tcsRNAinput\trnaseq\tannotation\tLog2Ratio vs. Input\tLog2Ratio vs. RNA\trev-csRNA\trev-csRNAinput\trev-RNA\tBidirectionalRatio\n",
      "chr8-54\tchr8\t23068947\t23069097\t-\t401.6\t0.871\t401.61\t188.71\t97.69\tna\t1.05818063625913\t1.95441205628819\t145.28\t124.90\t0.90\t1.41808147076813\n",
      "chr8-53\tchr8\t23068918\t23069068\t+\t400.8\t0.833\t400.78\t190.53\t6.97\tna\t1.04196758215362\t4.77117848609078\t58.45\t35.55\t67.82\t2.62098350798566\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t15v00_rna00/out.tss.txt <==\n",
      "#tssID\tchr\tstart\tend\tstrand\tscore\tfocusRatio\tcsRNA\tcsRNAinput\trnaseq\tannotation\tLog2Ratio vs. Input\tLog2Ratio vs. RNA\trev-csRNA\trev-csRNAinput\trev-RNA\tBidirectionalRatio\n",
      "chr8-102\tchr8\t23068909\t23069059\t-\t316.7\t0.578\t312.35\t145.98\t102.10\tna\t1.0569060147418\t1.54083452159989\t355.59\t212.05\t1.24\t-0.182663530402777\n",
      "chr8-148\tchr8\t23068918\t23069068\t+\t271.3\t0.795\t271.28\t122.54\t6.97\tna\t1.09722031758872\t4.22156613549434\t79.44\t53.28\t67.82\t1.67534695588085\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00_maxlen/out.tss.txt <==\n",
      "#tssID\tchr\tstart\tend\tstrand\tscore\tfocusRatio\tcsRNA\tcsRNAinput\trnaseq\tannotation\tLog2Ratio vs. Input\tLog2Ratio vs. RNA\trev-csRNA\trev-csRNAinput\trev-RNA\tBidirectionalRatio\n",
      "chr14-1\tchr14\t74302820\t74302970\t+\t1257.0\t0.900\t1257.02\t567.05\t0.32\tna\t1.13740088668881\t7.24836095044276\t22.41\t10.94\t0.42\t5.37847056755305\n",
      "chr14-2\tchr14\t74302859\t74303009\t-\t1254.3\t0.945\t1254.29\t567.96\t12.57\tna\t1.13200286872183\t5.93935779077944\t8.20\t36.47\t0.03\t6.28390577153239\n",
      "\n",
      "==> /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.tss.txt <==\n",
      "#tssID\tchr\tstart\tend\tstrand\tscore\tfocusRatio\tcsRNA\tcsRNAinput\trnaseq\tannotation\tLog2Ratio vs. Input\tLog2Ratio vs. RNA\trev-csRNA\trev-csRNAinput\trev-RNA\tBidirectionalRatio\n",
      "chr6-4\tchr6\t27139201\t27139351\t+\t989.7\t0.925\t988.23\t1.60\t0.63\tna\t6.69730064024309\t6.85097448667265\t4.73\t0.53\t0.50\t6.29017453195025\n",
      "chr19-7\tchr19\t33621872\t33622022\t+\t974.4\t0.977\t972.22\t120.94\t0.03\tna\t2.92640569459631\t6.93156178522793\t28.38\t0.53\t0.00\t4.75188813509376\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "head -3 /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss*/out.tss.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotation          Log2Ratio vs. Input Log2Ratio vs. RNA   rev-csRNA           rev-csRNAinput\n",
      "na                  6.69730064024309    6.85097448667265    4.73                0.53\n",
      "na                  2.92640569459631    6.93156178522793    28.38               0.53\n",
      "na                  1.32440745147403    5.54144207556596    5.46                27.71\n",
      "na                  1.34228540855552    6.8285789795433     21.83               10.12\n",
      "na                  1.03751421782274    6.85047796462051    11.64               36.23\n",
      "na                  1.04650882619672    6.86022653108661    5.82                3.73\n",
      "na                  2.98994890572271    6.86953215533549    13.46               1.07\n",
      "na                  1.13162085447592    6.76392369645699    145.18              252.54\n",
      "na                  1.13625477183514    6.79760603804027    85.87               93.77\n"
     ]
    }
   ],
   "source": [
    "!head /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.tss.txt | cut -f11-15 | expand -t 20"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "-----"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy  as np\n",
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "seq -> fastq -> trim/align -> BAM -> filtering (mapping quality)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tssID</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>focusRatio</th>\n",
       "      <th>csRNA</th>\n",
       "      <th>csRNAinput</th>\n",
       "      <th>rnaseq</th>\n",
       "      <th>annotation</th>\n",
       "      <th>Log2Ratio vs. Input</th>\n",
       "      <th>Log2Ratio vs. RNA</th>\n",
       "      <th>rev-csRNA</th>\n",
       "      <th>rev-csRNAinput</th>\n",
       "      <th>rev-RNA</th>\n",
       "      <th>BidirectionalRatio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr14-1</td>\n",
       "      <td>chr14</td>\n",
       "      <td>49862475</td>\n",
       "      <td>49862625</td>\n",
       "      <td>+</td>\n",
       "      <td>6671.1</td>\n",
       "      <td>0.974</td>\n",
       "      <td>6669.47</td>\n",
       "      <td>na</td>\n",
       "      <td>9.24</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>8.597398</td>\n",
       "      <td>0.00</td>\n",
       "      <td>na</td>\n",
       "      <td>0.16</td>\n",
       "      <td>9.705086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr14-2</td>\n",
       "      <td>chr14</td>\n",
       "      <td>49862774</td>\n",
       "      <td>49862924</td>\n",
       "      <td>-</td>\n",
       "      <td>6293.9</td>\n",
       "      <td>0.314</td>\n",
       "      <td>4910.20</td>\n",
       "      <td>na</td>\n",
       "      <td>4.73</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>8.593754</td>\n",
       "      <td>47.95</td>\n",
       "      <td>na</td>\n",
       "      <td>0.26</td>\n",
       "      <td>6.457848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr14-3</td>\n",
       "      <td>chr14</td>\n",
       "      <td>49586803</td>\n",
       "      <td>49586953</td>\n",
       "      <td>-</td>\n",
       "      <td>4523.4</td>\n",
       "      <td>0.887</td>\n",
       "      <td>4511.14</td>\n",
       "      <td>na</td>\n",
       "      <td>10.88</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>7.903046</td>\n",
       "      <td>0.00</td>\n",
       "      <td>na</td>\n",
       "      <td>0.00</td>\n",
       "      <td>9.141833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr17-1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>19187935</td>\n",
       "      <td>19188085</td>\n",
       "      <td>+</td>\n",
       "      <td>4359.3</td>\n",
       "      <td>0.945</td>\n",
       "      <td>4334.78</td>\n",
       "      <td>na</td>\n",
       "      <td>0.03</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>9.079003</td>\n",
       "      <td>0.00</td>\n",
       "      <td>na</td>\n",
       "      <td>0.00</td>\n",
       "      <td>9.084403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr19-1</td>\n",
       "      <td>chr19</td>\n",
       "      <td>33302460</td>\n",
       "      <td>33302610</td>\n",
       "      <td>-</td>\n",
       "      <td>4256.5</td>\n",
       "      <td>0.899</td>\n",
       "      <td>4254.86</td>\n",
       "      <td>na</td>\n",
       "      <td>1.77</td>\n",
       "      <td>na</td>\n",
       "      <td>na</td>\n",
       "      <td>8.769247</td>\n",
       "      <td>55.41</td>\n",
       "      <td>na</td>\n",
       "      <td>0.90</td>\n",
       "      <td>6.070967</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #tssID    chr     start       end strand   score  focusRatio    csRNA  \\\n",
       "0  chr14-1  chr14  49862475  49862625      +  6671.1       0.974  6669.47   \n",
       "1  chr14-2  chr14  49862774  49862924      -  6293.9       0.314  4910.20   \n",
       "2  chr14-3  chr14  49586803  49586953      -  4523.4       0.887  4511.14   \n",
       "3  chr17-1  chr17  19187935  19188085      +  4359.3       0.945  4334.78   \n",
       "4  chr19-1  chr19  33302460  33302610      -  4256.5       0.899  4254.86   \n",
       "\n",
       "  csRNAinput  rnaseq annotation Log2Ratio vs. Input  Log2Ratio vs. RNA  \\\n",
       "0         na    9.24         na                  na           8.597398   \n",
       "1         na    4.73         na                  na           8.593754   \n",
       "2         na   10.88         na                  na           7.903046   \n",
       "3         na    0.03         na                  na           9.079003   \n",
       "4         na    1.77         na                  na           8.769247   \n",
       "\n",
       "   rev-csRNA rev-csRNAinput  rev-RNA  BidirectionalRatio  \n",
       "0       0.00             na     0.16            9.705086  \n",
       "1      47.95             na     0.26            6.457848  \n",
       "2       0.00             na     0.00            9.141833  \n",
       "3       0.00             na     0.00            9.084403  \n",
       "4      55.41             na     0.90            6.070967  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath = \"/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t00/out.tss.txt\"\n",
    "dat = pd.read_csv(fpath, sep=\"\\t\")\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#tssID</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>score</th>\n",
       "      <th>focusRatio</th>\n",
       "      <th>csRNA</th>\n",
       "      <th>csRNAinput</th>\n",
       "      <th>rnaseq</th>\n",
       "      <th>annotation</th>\n",
       "      <th>Log2Ratio vs. Input</th>\n",
       "      <th>Log2Ratio vs. RNA</th>\n",
       "      <th>rev-csRNA</th>\n",
       "      <th>rev-csRNAinput</th>\n",
       "      <th>rev-RNA</th>\n",
       "      <th>BidirectionalRatio</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6-4</td>\n",
       "      <td>chr6</td>\n",
       "      <td>27139201</td>\n",
       "      <td>27139351</td>\n",
       "      <td>+</td>\n",
       "      <td>989.7</td>\n",
       "      <td>0.925</td>\n",
       "      <td>988.23</td>\n",
       "      <td>1.60</td>\n",
       "      <td>0.63</td>\n",
       "      <td>na</td>\n",
       "      <td>6.697301</td>\n",
       "      <td>6.850974</td>\n",
       "      <td>4.73</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.50</td>\n",
       "      <td>6.290175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr19-7</td>\n",
       "      <td>chr19</td>\n",
       "      <td>33621872</td>\n",
       "      <td>33622022</td>\n",
       "      <td>+</td>\n",
       "      <td>974.4</td>\n",
       "      <td>0.977</td>\n",
       "      <td>972.22</td>\n",
       "      <td>120.94</td>\n",
       "      <td>0.03</td>\n",
       "      <td>na</td>\n",
       "      <td>2.926406</td>\n",
       "      <td>6.931562</td>\n",
       "      <td>28.38</td>\n",
       "      <td>0.53</td>\n",
       "      <td>0.00</td>\n",
       "      <td>4.751888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr14-6</td>\n",
       "      <td>chr14</td>\n",
       "      <td>74302859</td>\n",
       "      <td>74303009</td>\n",
       "      <td>-</td>\n",
       "      <td>950.0</td>\n",
       "      <td>0.933</td>\n",
       "      <td>950.02</td>\n",
       "      <td>374.55</td>\n",
       "      <td>12.57</td>\n",
       "      <td>na</td>\n",
       "      <td>1.324407</td>\n",
       "      <td>5.541442</td>\n",
       "      <td>5.46</td>\n",
       "      <td>27.71</td>\n",
       "      <td>0.03</td>\n",
       "      <td>6.153305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr14-7</td>\n",
       "      <td>chr14</td>\n",
       "      <td>74302820</td>\n",
       "      <td>74302970</td>\n",
       "      <td>+</td>\n",
       "      <td>939.1</td>\n",
       "      <td>0.908</td>\n",
       "      <td>937.65</td>\n",
       "      <td>364.96</td>\n",
       "      <td>0.32</td>\n",
       "      <td>na</td>\n",
       "      <td>1.342285</td>\n",
       "      <td>6.828579</td>\n",
       "      <td>21.83</td>\n",
       "      <td>10.12</td>\n",
       "      <td>0.42</td>\n",
       "      <td>4.986470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8-1</td>\n",
       "      <td>chr8</td>\n",
       "      <td>135457375</td>\n",
       "      <td>135457525</td>\n",
       "      <td>+</td>\n",
       "      <td>933.6</td>\n",
       "      <td>0.961</td>\n",
       "      <td>933.65</td>\n",
       "      <td>450.74</td>\n",
       "      <td>0.16</td>\n",
       "      <td>na</td>\n",
       "      <td>1.037514</td>\n",
       "      <td>6.850478</td>\n",
       "      <td>11.64</td>\n",
       "      <td>36.23</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.583324</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #tssID    chr      start        end strand  score  focusRatio   csRNA  \\\n",
       "0   chr6-4   chr6   27139201   27139351      +  989.7       0.925  988.23   \n",
       "1  chr19-7  chr19   33621872   33622022      +  974.4       0.977  972.22   \n",
       "2  chr14-6  chr14   74302859   74303009      -  950.0       0.933  950.02   \n",
       "3  chr14-7  chr14   74302820   74302970      +  939.1       0.908  937.65   \n",
       "4   chr8-1   chr8  135457375  135457525      +  933.6       0.961  933.65   \n",
       "\n",
       "   csRNAinput  rnaseq annotation  Log2Ratio vs. Input  Log2Ratio vs. RNA  \\\n",
       "0        1.60    0.63         na             6.697301           6.850974   \n",
       "1      120.94    0.03         na             2.926406           6.931562   \n",
       "2      374.55   12.57         na             1.324407           5.541442   \n",
       "3      364.96    0.32         na             1.342285           6.828579   \n",
       "4      450.74    0.16         na             1.037514           6.850478   \n",
       "\n",
       "   rev-csRNA  rev-csRNAinput  rev-RNA  BidirectionalRatio  \n",
       "0       4.73            0.53     0.50            6.290175  \n",
       "1      28.38            0.53     0.00            4.751888  \n",
       "2       5.46           27.71     0.03            6.153305  \n",
       "3      21.83           10.12     0.42            4.986470  \n",
       "4      11.64           36.23     0.00            5.583324  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fpath = \"/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/out_findTSS/tss_t60v00_rna00/out.tss.txt\"\n",
    "dat = pd.read_csv(fpath, sep=\"\\t\")\n",
    "dat.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Monday: 3pm"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "a lot of bioinformatics post analysis on Github \n",
    "BioSTAR\n",
    "Google Group\n",
    "\n",
    "Deeptools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['na'], dtype=object)"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat['annotation'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>csRNA</th>\n",
       "      <th>csRNAinput</th>\n",
       "      <th>rnaseq</th>\n",
       "      <th>Log2Ratio vs. Input</th>\n",
       "      <th>Log2Ratio vs. RNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>972.22</td>\n",
       "      <td>120.94</td>\n",
       "      <td>0.03</td>\n",
       "      <td>2.926406</td>\n",
       "      <td>6.931562</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>950.02</td>\n",
       "      <td>374.55</td>\n",
       "      <td>12.57</td>\n",
       "      <td>1.324407</td>\n",
       "      <td>5.541442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>937.65</td>\n",
       "      <td>364.96</td>\n",
       "      <td>0.32</td>\n",
       "      <td>1.342285</td>\n",
       "      <td>6.828579</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    csRNA  csRNAinput  rnaseq  Log2Ratio vs. Input  Log2Ratio vs. RNA\n",
       "1  972.22      120.94    0.03             2.926406           6.931562\n",
       "2  950.02      374.55   12.57             1.324407           5.541442\n",
       "3  937.65      364.96    0.32             1.342285           6.828579"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.loc[1:3, [\"csRNA\", \"csRNAinput\", \"rnaseq\", \"Log2Ratio vs. Input\", \"Log2Ratio vs. RNA\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0        9.270631\n",
       "1        3.006991\n",
       "2        1.342800\n",
       "3        1.361311\n",
       "4        1.050586\n",
       "           ...   \n",
       "23766         inf\n",
       "23767         inf\n",
       "23768         inf\n",
       "23769         inf\n",
       "23770         inf\n",
       "Length: 23771, dtype: float64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.log2(dat[\"csRNA\"] / dat[\"csRNAinput\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:>"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD7CAYAAAB68m/qAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuNCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8QVMy6AAAACXBIWXMAAAsTAAALEwEAmpwYAAAQZ0lEQVR4nO3dbYylZ13H8e+PLQh2oLxYnDS7lW3iSmxYY9tJC6nBGXnIFkjXhEZaSnUJZd+wBNOqWcQUrG9AgfhU0U1bAQUmWAQ3dqUY6QRNLLbL07Jbata62l2R5XF1KloX/r6Y0+UwnZlzdubMnDnXfD/JZu+H65zzvzLJ71xz3dd9T6oKSdLoe8qwC5AkDYaBLkmNMNAlqREGuiQ1wkCXpEYY6JLUiJ6BnuSuJKeSfGmR80nye0mOJflikssGX6YkqZd+RujvA3Yucf5qYHvn3x7gvSsvS5J0rs7r1aCqPp1k2xJNdgEfqLk7lO5P8uwkF1bVV5Z6382bN9e2bUu97eIee+wxzj///GW9dj1rsV8t9gna7Jd9Gg2HDh36elU9Z6FzPQO9D1uAR7v2T3SOPSnQk+xhbhTP+Pg473rXu5b1gbOzs4yNjS3rtetZi/1qsU/QZr/s02iYmpr618XODSLQ+1ZV+4H9ABMTEzU5Obms95mZmWG5r13PWuxXi32CNvtln0bfIFa5nAQu6trf2jkmSVpDgwj0A8AvdFa7vAA43Wv+XJI0eD2nXJJ8GJgENic5AbwNeCpAVf0RcBB4OXAM+G/gdatVrCRpcf2scrm+x/kC3jiwiiRJy+KdopLUCANdkhphoEtSIwx0SWrEmt5YNKq27bvn7Pbxd7xiiJVI0uIcoUtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUCANdkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNaKvQE+yM8nDSY4l2bfA+R9Ncl+SzyX5YpKXD75USdJSegZ6kk3A7cDVwCXA9Ukumdfs14GPVNWlwHXAHw66UEnS0voZoV8BHKuqR6rqcWAa2DWvTQHP6mxfAPz74EqUJPXjvD7abAEe7do/AVw5r83bgU8meRNwPvCSgVQnSepbqmrpBsm1wM6quqmzfyNwZVXt7Wpzc+e93p3khcCdwPOr6nvz3msPsAdgfHz88unp6WUVPTs7y9jY2LJeuxyHT54+u71jywWr9jlr3a+10GKfoM1+2afRMDU1daiqJhY6188I/SRwUdf+1s6xbq8HdgJU1T8keTqwGTjV3aiq9gP7ASYmJmpycrKf+p9kZmaG5b52OXbvu+fs9vEbVu9z17pfa6HFPkGb/bJPo6+fQH8A2J7kYuaC/DrgNfPa/BvwYuB9SX4CeDrwtUEWuh5t6w76d7xiiJVIUh8XRavqDLAXuBd4iLnVLEeS3Jbkmk6zW4A3JPkC8GFgd/Way5EkDVQ/I3Sq6iBwcN6xW7u2jwJXDbY0SdK58E5RSWqEgS5JjTDQJakRBrokNcJAl6RG9LXKRd/XvfZcktYTR+iS1AgDXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQI7xQdEP96kaRhc4QuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGuGyxVXgEkZJw+AIXZIaYaBLUiMMdElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIbyxaRPfNQZI0Cgz0VeZdo5LWilMuktQIA12SGtFXoCfZmeThJMeS7Fukzc8nOZrkSJIPDbZMSVIvPefQk2wCbgdeCpwAHkhyoKqOdrXZDrwFuKqqvpXkR1arYEnSwvoZoV8BHKuqR6rqcWAa2DWvzRuA26vqWwBVdWqwZUqSeklVLd0guRbYWVU3dfZvBK6sqr1dbT4O/BNwFbAJeHtVfWKB99oD7AEYHx+/fHp6ellFz87OMjY2tqzX9uvwydMDf88dWy5Y8vxa9GuttdgnaLNf9mk0TE1NHaqqiYXODWrZ4nnAdmAS2Ap8OsmOqvp2d6Oq2g/sB5iYmKjJycllfdjMzAzLfe1SfnDt+eBXdB6/YXLJ86vVr2FqsU/QZr/s0+jrZ8rlJHBR1/7WzrFuJ4ADVfV/VfUvzI3Wtw+mRElSP/oJ9AeA7UkuTvI04DrgwLw2H2dudE6SzcCPA48MrkxJUi895xWq6kySvcC9zM2P31VVR5LcBjxYVQc6516W5CjwXeBXquobq1n4qPMOUkmD1tdEcVUdBA7OO3Zr13YBN3f+SZKGwDtFJakRPpxrDfkER0mryRG6JDXCQJekRhjoktQIA12SGmGgS1IjXOWyDjyx+uWWHWfY7Q1HkpbJEbokNcJAl6RGGOiS1AgDXZIaYaBLUiM2/CoXn68iqRWO0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJasSGX4c+Krb5FEZJPWzIQPdmIkktcspFkhphoEtSIwx0SWqEgS5JjTDQJakRBrokNcJAl6RGbIh16K47l7QRbIhAb5l3kEp6glMuktQIR+jrmFNFks5FXyP0JDuTPJzkWJJ9S7R7VZJKMjG4EiVJ/egZ6Ek2AbcDVwOXANcnuWSBds8E3gx8ZtBFSpJ662eEfgVwrKoeqarHgWlg1wLtfhN4J/A/A6xPktSnVNXSDZJrgZ1VdVNn/0bgyqra29XmMuCtVfWqJDPAL1fVgwu81x5gD8D4+Pjl09PTyyp6dnaWsbGxvtsfPnl6WZ+z1safAV/9Tu92O7ZccHa7u2/dx9eLc/1ZjYoW+2WfRsPU1NShqlpwWnvFF0WTPAV4D7C7V9uq2g/sB5iYmKjJycllfebMzAzn8trdI3Jx8ZYdZ3j34d4/kuM3TJ7d7u5b9/H14lx/VqOixX7Zp9HXz5TLSeCirv2tnWNPeCbwfGAmyXHgBcABL4xK0trqJ9AfALYnuTjJ04DrgANPnKyq01W1uaq2VdU24H7gmoWmXCRJq6fn7/dVdSbJXuBeYBNwV1UdSXIb8GBVHVj6HYaj5TXcLfdN0vL1NYdeVQeBg/OO3bpI28mVlyVJOlfe+i9JjRjJQD988jTb9t3j1IMkdRnJQJckPdnIP5zLx8dK0hxH6JLUiJEfoXdzTl3SRtZUoOv7nIqSNh6nXCSpEQa6JDXCQJekRjiH3hAvCksbmyN0SWqEgS5JjTDQJakRzqFvMK5Pl9rlCF2SGmGgS1IjDHRJaoSBLkmNMNAlqRGuctkAvINU2hgcoUtSIxyhb2DzR+6uS5dGmyN0SWqEgS5JjXDKRWf5WABptDlCl6RGGOiS1AgDXZIaYaBLUiMMdElqhKtc1JOrX6TR4AhdkhphoEtSI/oK9CQ7kzyc5FiSfQucvznJ0SRfTPK3SZ47+FIlSUvpGehJNgG3A1cDlwDXJ7lkXrPPARNV9ZPA3cBvDbpQSdLS+rkoegVwrKoeAUgyDewCjj7RoKru62p/P/DaQRap9cmLpdL60k+gbwEe7do/AVy5RPvXA3+9kqI0fP5RDGn0pKqWbpBcC+ysqps6+zcCV1bV3gXavhbYC/xMVf3vAuf3AHsAxsfHL5+enl5W0ae+eZqvfmdZL13Xxp/Buu/Xji0XnN0+fPL0gse7zc7OMjY2tup1rbUW+2WfRsPU1NShqppY6Fw/I/STwEVd+1s7x35AkpcAb2WRMAeoqv3AfoCJiYmanJzs4+Of7Pc/+Je8+3B7S+hv2XFm3ffr+A2TZ7d3d0+5dB3vNjMzw3J/zutZi/2yT6Ovn/R4ANie5GLmgvw64DXdDZJcCvwxcyP5UwOvUuuGUzHS+tVzlUtVnWFuGuVe4CHgI1V1JMltSa7pNPttYAz48ySfT3Jg1SqWJC2or9/vq+ogcHDesVu7tl8y4LrUiMMnT5+dmnEljLS6vFNUkhqxvq/AaWS4Jl0aPkfoktQIR+gauO7R+i07hliItMEY6FozTstIq8spF0lqhIEuSY1wykVD4fSLNHiO0CWpEQa6JDXCQJekRhjoktQIL4pq6BZ7JK8XS6Vz4whdkhrhCF3rlksbpXPjCF2SGuEIXSOhn9G6I3ptdI7QJakRjtA1chyJSwsz0NU8vwC0UTjlIkmNMNAlqREGuiQ1wjl0jbTFHhvQz3Hn09UaR+iS1AhH6NqwHK2rNY7QJakRjtClHhaaj79lxxkm174UaUkGusTKp1+cvtF6YKBL8yy2Qma57aS1YqBLA+ZoXcPiRVFJaoQjdGkVneto3dG9VsJAl9aIYa3VZqBLQ9DPBVW/AHSu+gr0JDuB3wU2AXdU1Tvmnf8h4APA5cA3gFdX1fHBliptXIt9AXQH/WJfAH4xbBw9Az3JJuB24KXACeCBJAeq6mhXs9cD36qqH0tyHfBO4NWrUbCk71vJw8m6LfYFsFib+e38olgf+hmhXwEcq6pHAJJMA7uA7kDfBby9s3038AdJUlU1wFolrZJt++7hlh1n2L3EVNBS00TnuiZ/NX6D8I5eSK/MTXItsLOqburs3whcWVV7u9p8qdPmRGf/nzttvj7vvfYAezq7zwMeXmbdm4Gv92w1elrsV4t9gjb7ZZ9Gw3Or6jkLnVjTi6JVtR/Yv9L3SfJgVU0MoKR1pcV+tdgnaLNf9mn09XNj0Ungoq79rZ1jC7ZJch5wAXMXRyVJa6SfQH8A2J7k4iRPA64DDsxrcwD4xc72tcCnnD+XpLXVc8qlqs4k2Qvcy9yyxbuq6kiS24AHq+oAcCfwp0mOAd9kLvRX04qnbdapFvvVYp+gzX7ZpxHX86KoJGk0+HAuSWqEgS5JjRipQE9yV5JTnXXvTUhyUZL7khxNciTJm4dd0yAkeXqSf0zyhU6/fmPYNQ1Kkk1JPpfkr4Zdy6AkOZ7kcJLPJ3lw2PUMQpJnJ7k7yZeTPJTkhcOuabWN1Bx6khcBs8AHqur5w65nEJJcCFxYVZ9N8kzgEPBz8x6tMHKSBDi/qmaTPBX4e+DNVXX/kEtbsSQ3AxPAs6rqlcOuZxCSHAcm5t8MOMqSvB/4u6q6o7NC74er6ttDLmtVjdQIvao+zdwqmmZU1Veq6rOd7f8CHgK2DLeqlas5s53dp3b+jc7oYRFJtgKvAO4Ydi1aXJILgBcxtwKPqnq89TCHEQv01iXZBlwKfGbIpQxEZ2ri88Ap4G+qqoV+/Q7wq8D3hlzHoBXwySSHOo/oGHUXA18D/qQzPXZHkvOHXdRqM9DXiSRjwEeBX6qq/xx2PYNQVd+tqp9i7u7iK5KM9DRZklcCp6rq0LBrWQU/XVWXAVcDb+xMb46y84DLgPdW1aXAY8C+4Za0+gz0daAzx/xR4INV9RfDrmfQOr/q3gfsHHIpK3UVcE1nvnka+Nkkfzbckgajqk52/j8FfIy5p6yOshPAia7fCu9mLuCbZqAPWefi4Z3AQ1X1nmHXMyhJnpPk2Z3tZzD3PP0vD7WoFaqqt1TV1qraxtzd0J+qqtcOuawVS3J+54I8nWmJlwEjvZKsqv4DeDTJ8zqHXswPPvK7SSP1J+iSfBiYBDYnOQG8raruHG5VK3YVcCNwuDPfDPBrVXVweCUNxIXA+zt/IOUpwEeqqpllfo0ZBz42N7bgPOBDVfWJ4ZY0EG8CPthZ4fII8Loh17PqRmrZoiRpcU65SFIjDHRJaoSBLkmNMNAlqREGuiQ1wkCXpEYY6JLUiP8H2z/5/l2yU78AAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "filenames": {
       "image/png": "/gpfs/fs1/data/reddylab/Kuei/GitRepo/Proj_PROcap_analysis/document/_build/jupyter_execute/notebooks/12_run_homer_procap/04_homer_findTSS_49_1.png"
      },
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat[\"Log2Ratio vs. Input\"].hist(cumulative=False, density=1, bins=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<AxesSubplot:>"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAXQAAAD4CAYAAAD8Zh1EAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuNCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8QVMy6AAAACXBIWXMAAAsTAAALEwEAmpwYAAAQxUlEQVR4nO3df4xlZX3H8ffHRSpl7Npk7YSw1CXp1pSwTYEJ1NDobJVmsQaaSBSCpBhx+4fbaNz+wNqgpf1D22DaRmq7QeuPKhPE2mzoRmyUibUpdll/rSzSrHStu1XwJ3aQFrd++8fcXYdh2Llz5+zee595v5IJ59zz3DPfZyAfvvOcc+6kqpAkjb9nDLsASVI3DHRJaoSBLkmNMNAlqREGuiQ14rRhfeMNGzbUpk2bBnrvY489xplnntltQUPS0lygrfk4l9HV0nxWOpd9+/Z9q6qeu9SxoQX6pk2buO+++wZ67+zsLNPT090WNCQtzQXamo9zGV0tzWelc0ny1ac75pKLJDXCQJekRhjoktQIA12SGmGgS1IjDHRJasSygZ7kPUkeSfKlpzmeJH+Z5GCSLya5sPsyJUnL6adDfy+w7QTHLwc29762A+9afVmSpJVaNtCr6lPAd04w5Erg/TXvXuA5Sc7qqkBJUn/Szx+4SLIJuKuqzl/i2F3A26rq0739TwC/X1VPeQw0yXbmu3gmJycvmpmZGajoubk5JiYmBnrvqGlpLtDWfJxLt/YfebSzc02eAQ8/3tnpTrktZ68/vr3Sfzdbt27dV1VTSx07pY/+V9UuYBfA1NRUDfro7lp+7HfUtTSftTSXTTf+4ymooru42bnlKLfsH9onl6zaoWunj293+d9ZFz+RI8A5C/Y39l6TNCSLA3rnlqNcf0pCW8PURaDvBnYkmQEuAR6tqq93cF5JSzg13bTG0bKBnuR2YBrYkOQw8BbgmQBV9dfAHuClwEHgB8CrT1ax0lpicGullg30qrpmmeMFvK6ziqQ1xuBWV8b3qoI0ZgxunWwGunQSGeI6lQx0qQMGt0aBgS4NyBDXqDHQpRUwxDXKDHRpGYa4xoWBLi1yLMB9ulLjxj9wIUmNsEOXcFlFbTDQtWYZ4mqNga41xRBXywx0Nc8Q11rhRVFJaoQduppkV661yEBXMwxxrXUuuUhSI+zQNdbsyqUfs0OXpEbYoWvs2JVLS7NDl6RG2KFrLNiVS8uzQ5ekRtiha2TZlUsrY4cuSY2wQ9dIsSuXBmeHLkmNsEPX0NmVS92wQ5ekRhjoktQIl1w0FC6zSN2zQ5ekRtih65SxK5dOrr469CTbkjyY5GCSG5c4/rNJ7knyuSRfTPLS7kuVJJ3IsoGeZB1wK3A5cB5wTZLzFg37Q+COqroAuBr4q64LlSSdWD9LLhcDB6vqIYAkM8CVwIEFYwr4qd72euC/uixS48tlFunUSVWdeEByFbCtqm7o7V8HXFJVOxaMOQv4OPDTwJnAS6pq3xLn2g5sB5icnLxoZmZmoKLn5uaYmJgY6L2jpqW5wFPns//Io0OsZnUmz4CHHx92Fd1oaS4w/vPZcvb649srzYCtW7fuq6qppY51dVH0GuC9VXVLkhcAH0hyflX9aOGgqtoF7AKYmpqq6enpgb7Z7Owsg7531LQ0F3jqfK4f4w5955aj3LK/jfsGWpoLjP98Dl07fXy7ywzo5ydyBDhnwf7G3msLvQbYBlBV/5rkWcAG4JEuitR4cZlFGo5+7nLZC2xOcm6S05m/6Ll70Zj/BF4MkOQXgGcB3+yyUEnSiS0b6FV1FNgB3A08wPzdLPcnuTnJFb1hO4HXJvkCcDtwfS23OC9J6lRfi1BVtQfYs+i1mxZsHwAu7bY0jZNjyyw7txzF59Wk4fDRf0lqhK2UBubFT2m02KFLUiMMdElqhEsuWhGXWaTRZYcuSY0w0CWpEQa6JDXCNXQty3VzaTzYoUtSIwx0SWqESy5aksss0vixQ5ekRhjoktQIl1x0nMss0nizQ5ekRhjoktQIA12SGuEa+hrmmrnUFjt0SWqEgS5JjTDQJakRrqGvMa6bS+2yQ5ekRhjoktQIA12SGuEa+hrgurm0NtihS1IjDHRJaoSBLkmNcA29Ua6bS2uPHbokNaKvQE+yLcmDSQ4mufFpxrwiyYEk9yf5ULdlSpKWs+ySS5J1wK3AZcBhYG+S3VV1YMGYzcCbgEur6rtJfuZkFSxJWlo/HfrFwMGqeqiqngBmgCsXjXktcGtVfRegqh7ptkxJ0nL6uSh6NvC1BfuHgUsWjfl5gCT/AqwD3lpVH+ukQvXNC6HS2paqOvGA5CpgW1Xd0Nu/DrikqnYsGHMX8EPgFcBG4FPAlqr63qJzbQe2A0xOTl40MzMzUNFzc3NMTEwM9N5R0+Vc9h95tJPzrMbkGfDw48OuohvOZXSN+3y2nL3++PZKM2Dr1q37qmpqqWP9dOhHgHMW7G/svbbQYeAzVfVD4D+S/DuwGdi7cFBV7QJ2AUxNTdX09HRfE1hsdnaWQd87arqcy/Uj0KHv3HKUW/a3cTescxld4z6fQ9dOH9/uMgP6WUPfC2xOcm6S04Grgd2LxvwDMA2QZAPzSzAPdVKhJKkvywZ6VR0FdgB3Aw8Ad1TV/UluTnJFb9jdwLeTHADuAX63qr59soqWJD1VX7+zVNUeYM+i125asF3AG3tfOoW8ECrpGJ8UlaRGGOiS1AgDXZIaYaBLUiPG90bONcwLoZKWYocuSY0w0CWpEQa6JDXCQJekRnhRdEx4IVTScuzQJakRBrokNcJAl6RGGOiS1AgDXZIa4V0uI8w7WySthB26JDXCQJekRhjoktQIA12SGuFF0RHjhVBJg7JDl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY3wtsUR4K2Kkrpghy5JjTDQJakRBrokNcJAl6RGGOiS1Ii+Aj3JtiQPJjmY5MYTjHt5kkoy1V2JkqR+LHvbYpJ1wK3AZcBhYG+S3VV1YNG4ZwOvBz5zMgptzbFbFXduOYp3j0rqQj8d+sXAwap6qKqeAGaAK5cY98fA24H/6bA+SVKfUlUnHpBcBWyrqht6+9cBl1TVjgVjLgTeXFUvTzIL/E5V3bfEubYD2wEmJycvmpmZGajoubk5JiYmBnrvqNh/5FEAJs+Ahx8fcjEdamk+zmV0jft8tpy9/vj2SvNs69at+6pqyWXtVf+un+QZwDuA65cbW1W7gF0AU1NTNT09PdD3nJ2dZdD3jorrFyy53LK/nSWXlubjXEbXuM/n0LXTx7e7zLN+llyOAOcs2N/Ye+2YZwPnA7NJDgG/DOz2wqgknVr9BPpeYHOSc5OcDlwN7D52sKoeraoNVbWpqjYB9wJXLLXkIkk6eZb9naWqjibZAdwNrAPeU1X3J7kZuK+qdp/4DDrGD+GSdDL1tQhVVXuAPYteu+lpxk6vvixJ0kr5pKgkNcJAl6RGGOiS1AgDXZIaYaBLUiPG91GrMeGtipJOFTt0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AgDXZIa4X3oJ4H3nksaBjt0SWqEgS5JjTDQJakRBrokNcJAl6RGGOiS1AhvW+yItypKGjY7dElqhIEuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGuF96KvgveeSRokduiQ1wkCXpEYY6JLUiL4CPcm2JA8mOZjkxiWOvzHJgSRfTPKJJM/rvlRJ0oksG+hJ1gG3ApcD5wHXJDlv0bDPAVNV9YvAncCfdl2oJOnE+unQLwYOVtVDVfUEMANcuXBAVd1TVT/o7d4LbOy2TEnSclJVJx6QXAVsq6obevvXAZdU1Y6nGf9O4BtV9SdLHNsObAeYnJy8aGZmZqCi5+bmmJiYGOi9Xdp/5NFVn2PyDHj48Q6KGREtzce5jK5xn8+Ws9cf315pnm3dunVfVU0tdazT+9CTvAqYAl601PGq2gXsApiamqrp6emBvs/s7CyDvne1nnzv+ep/fDu3HOWW/e08DtDSfJzL6Br3+Ry6dvr4dpd51s9P5AhwzoL9jb3XniTJS4A3Ay+qqv/tpDpJUt/6WUPfC2xOcm6S04Grgd0LByS5APgb4IqqeqT7MiVJy1k20KvqKLADuBt4ALijqu5PcnOSK3rD/gyYAD6c5PNJdj/N6SRJJ0lfi1BVtQfYs+i1mxZsv6TjuiRJK+STopLUCANdkhphoEtSIwx0SWqEgS5JjRjfR61OIf8ykaRxYIcuSY0w0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjDHRJaoQPFj0NHyaSNG7s0CWpEQa6JDXCQJekRhjoktQIA12SGmGgS1IjvG1xAW9VlDTO7NAlqREGuiQ1wkCXpEYY6JLUiDV/UdQLoZJaYYcuSY0w0CWpEWtyycVlFkktskOXpEasiQ7djlzSWmCHLkmN6KtDT7IN+AtgHXBbVb1t0fGfAN4PXAR8G3hlVR3qttSVsSuXtNYsG+hJ1gG3ApcBh4G9SXZX1YEFw14DfLeqfi7J1cDbgVeejIJPxBCXtJb106FfDBysqocAkswAVwILA/1K4K297TuBdyZJVVWHtR63/8ijXG94S9KTZLnMTXIVsK2qbujtXwdcUlU7Foz5Um/M4d7+V3pjvrXoXNuB7b3d5wMPDlj3BuBby44aDy3NBdqaj3MZXS3NZ6VzeV5VPXepA6f0Lpeq2gXsWu15ktxXVVMdlDR0Lc0F2pqPcxldLc2ny7n0c5fLEeCcBfsbe68tOSbJacB65i+OSpJOkX4CfS+wOcm5SU4HrgZ2LxqzG/jN3vZVwCdP1vq5JGlpyy65VNXRJDuAu5m/bfE9VXV/kpuB+6pqN/Bu4ANJDgLfYT70T6ZVL9uMkJbmAm3Nx7mMrpbm09lclr0oKkkaDz4pKkmNMNAlqRFjFehJ3pPkkd5972MtyTlJ7klyIMn9SV4/7JoGleRZSf4tyRd6c/mjYde0WknWJflckruGXctqJTmUZH+Szye5b9j1rEaS5yS5M8mXkzyQ5AXDrmlQSZ7f+3dy7Ov7Sd6wqnOO0xp6khcCc8D7q+r8YdezGknOAs6qqs8meTawD/iNRR+pMBaSBDizquaSPBP4NPD6qrp3yKUNLMkbgSngp6rqZcOuZzWSHAKmFj/oN46SvA/456q6rXfX3U9W1feGXNaq9T5i5QjzD2R+ddDzjFWHXlWfYv4umrFXVV+vqs/2tv8beAA4e7hVDabmzfV2n9n7Gp9OYZEkG4FfB24bdi36sSTrgRcyf1cdVfVEC2He82LgK6sJcxizQG9Vkk3ABcBnhlzKwHpLFJ8HHgH+qarGdi7AnwO/B/xoyHV0pYCPJ9nX+/iNcXUu8E3gb3vLYbclOXPYRXXkauD21Z7EQB+yJBPAR4A3VNX3h13PoKrq/6rql5h/kvjiJGO5JJbkZcAjVbVv2LV06Feq6kLgcuB1vaXLcXQacCHwrqq6AHgMuHG4Ja1eb+noCuDDqz2XgT5EvfXmjwAfrKq/H3Y9Xej9CnwPsG3IpQzqUuCK3rrzDPCrSf5uuCWtTlUd6f3zEeCjzH+C6jg6DBxe8NvfncwH/Li7HPhsVT282hMZ6EPSu5D4buCBqnrHsOtZjSTPTfKc3vYZzH92/peHWtSAqupNVbWxqjYx/2vwJ6vqVUMua2BJzuxddKe3PPFrwFjeJVZV3wC+luT5vZdezJM/xntcXUMHyy0wZn9TNMntwDSwIclh4C1V9e7hVjWwS4HrgP29tWeAP6iqPcMraWBnAe/rXal/BnBHVY397X6NmAQ+Ot8/cBrwoar62HBLWpXfBj7YW6Z4CHj1kOtZld7/ZC8DfquT843TbYuSpKfnkoskNcJAl6RGGOiS1AgDXZIaYaBLUiMMdElqhIEuSY34fxxk9m3trb2bAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "filenames": {
       "image/png": "/gpfs/fs1/data/reddylab/Kuei/GitRepo/Proj_PROcap_analysis/document/_build/jupyter_execute/notebooks/12_run_homer_procap/04_homer_findTSS_50_1.png"
      },
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "dat[\"Log2Ratio vs. RNA\"].hist(cumulative=True, density=1, bins=100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}