
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>2.2. Run HOMER makeTagDirectory of PRO-cap and RNA-seq data &#8212; My sample book</title>
    
  <link rel="stylesheet" href="../../_static/css/index.f658d18f9b420779cfdf24aa0a7e2d77.css">

    
  <link rel="stylesheet"
    href="../../_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="../../_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="../../_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="../../_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/sphinx-book-theme.e7340bb3dbd8dde6db86f25597f54a1b.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="../../_static/js/index.d3f166471bb80abb5163.js">

    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <script src="../../_static/togglebutton.js"></script>
    <script src="../../_static/clipboard.min.js"></script>
    <script src="../../_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="../../_static/sphinx-book-theme.7d483ff0a819d6edff12ce0b1ead3928.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="../../_static/sphinx-thebe.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="2.3. Create Genome Browser Visualization Files" href="03_homer_vis.html" />
    <link rel="prev" title="2.1. Check the file paths for the data" href="01_prep_files.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="../../index.html">
  
  <img src="../../_static/logo.png" class="logo" alt="logo">
  
  
  <h1 class="site-logo" id="site-title">My sample book</h1>
  
</a>
</div><form class="bd-search d-flex align-items-center" action="../../search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>
<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
    <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../../intro.html">
   Welcome to your Jupyter Book
  </a>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Example
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1 collapsible-parent">
  <a class="reference internal" href="../../content.html">
   Content in Jupyter Book
  </a>
  <ul class="collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="../../markdown.html">
     Markdown Files
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="../../notebooks.html">
     Content with notebooks
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>
<p class="caption collapsible-parent">
 <span class="caption-text">
  Peak calling of PRO-cap data
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="../11_prep_procap/01_prep_files.html">
   1. PRO-cap Data description and preparation
  </a>
 </li>
 <li class="toctree-l1 current active collapsible-parent">
  <a class="reference internal" href="00_homer_intro.html">
   2. Run Homer Analysis on PRO-cap Data
  </a>
  <ul class="current collapse-ul">
   <li class="toctree-l2">
    <a class="reference internal" href="01_prep_files.html">
     2.1. Check the file paths for the data
    </a>
   </li>
   <li class="toctree-l2 current active">
    <a class="current reference internal" href="#">
     2.2. Run HOMER makeTagDirectory of PRO-cap and RNA-seq data
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="03_homer_vis.html">
     2.3. Create Genome Browser Visualization Files
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="04_homer_findPeaks.html">
     2.4. HOMER FindPeaks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="04_homer_findTSS.html">
     2.5. HOMER FindcsRNATSS
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="05_homer_merge_peak.html">
     2.6. Merge Peak output from HOMER findPeaks
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="05_homer_merge_tss.html">
     2.7. Merge TSS output from HOMER findcsRNATSS
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="06_homer_annot_peak.html">
     2.8. Get the count using HOMER annotatePeaks.pl
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="06_homer_annot_tss.html">
     2.9. Get the count using HOMER annotatePeaks.pl
    </a>
   </li>
   <li class="toctree-l2">
    <a class="reference internal" href="12_peak_PER1.html">
     2.10. Summarize and visualize peaks from HOMER pipeline
    </a>
   </li>
  </ul>
  <i class="fas fa-chevron-down">
  </i>
 </li>
</ul>

</nav> <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="topbar container-xl fixed-top">
    <div class="topbar-contents row">
        <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show"></div>
        <div class="col pl-md-4 topbar-main">
            
            <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
                data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
                aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
                title="Toggle navigation" data-toggle="tooltip" data-placement="left">
                <i class="fas fa-bars"></i>
                <i class="fas fa-arrow-left"></i>
                <i class="fas fa-arrow-up"></i>
            </button>
            
            
<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="../../_sources/notebooks/12_run_homer_procap/02_homer_tag.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.ipynb</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
</div>

            <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/executablebooks/jupyter-book"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Fnotebooks/12_run_homer_procap/02_homer_tag.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        
    </div>
</div>


            <!-- Full screen (wrap in <a> to have style consistency -->
            <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                    data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                    title="Fullscreen mode"><i
                        class="fas fa-expand"></i></button></a>

            <!-- Launch buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Launch interactive content"><i class="fas fa-rocket"></i></button>
    <div class="dropdown-buttons">
        
        <a class="binder-button" href="https://mybinder.org/v2/gh/executablebooks/jupyter-book/master?urlpath=tree/docs/notebooks/12_run_homer_procap/02_homer_tag.ipynb"><button type="button"
                class="btn btn-secondary topbarbtn" title="Launch Binder" data-toggle="tooltip"
                data-placement="left"><img class="binder-button-logo"
                    src="../../_static/images/logo_binder.svg"
                    alt="Interact on binder">Binder</button></a>
        
        
        
        
    </div>
</div>

        </div>

        <!-- Table of contents -->
        <div class="d-none d-md-block col-md-2 bd-toc show">
            
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i>
            Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#homer-maketagdirectory-of-single-sample-option-genome-hg38">
   2.2.1. HOMER makeTagDirectory of single sample (Option: genome hg38)
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#quality-control-checks">
     2.2.1.1. Quality Control Checks
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#homer-maketagdirectory-of-single-sample-option-maxlen-65">
   2.2.2. HOMER makeTagDirectory of single sample (Option: maxlen 65)
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id1">
     2.2.2.1. Quality Control Checks:
    </a>
    <ul class="nav section-nav flex-column">
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#read-clonality-pcr-duplicates">
       2.2.2.1.1. Read clonality/PCR duplicates
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#read-length-distribution">
       2.2.2.1.2. Read length distribution
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#nucleotide-preferences-near-the-5-end-of-the-read">
       2.2.2.1.3. Nucleotide preferences near the 5’ end of the read
      </a>
     </li>
     <li class="toc-h4 nav-item toc-entry">
      <a class="reference internal nav-link" href="#distribution-of-reads-relative-to-one-another">
       2.2.2.1.4. Distribution of reads relative to one another
      </a>
     </li>
    </ul>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#run-maketagdirectory-for-all-sample">
   2.2.3. Run makeTagDirectory for all sample
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#pro-cap-all-samples">
     2.2.3.1. PRO-cap (all samples)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#pro-cap-all-samples-options-maxlen">
     2.2.3.2. PRO-cap (all samples; options: maxlen)
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#rna-seq-3-time-points">
     2.2.3.3. RNA-seq (3 time points)
    </a>
   </li>
  </ul>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#check-output-message">
   2.2.4. Check output message
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#id2">
   2.2.5. Quality Control Checks
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id3">
     2.2.5.1. Read clonality/PCR duplicates
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id4">
     2.2.5.2. Read length distribution
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#nucleotide-preferences">
     2.2.5.3. Nucleotide preferences
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#id5">
     2.2.5.4. Distribution of reads relative to one another
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
        </div>
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="run-homer-maketagdirectory-of-pro-cap-and-rna-seq-data">
<h1><span class="section-number">2.2. </span>Run HOMER makeTagDirectory of PRO-cap and RNA-seq data<a class="headerlink" href="#run-homer-maketagdirectory-of-pro-cap-and-rna-seq-data" title="Permalink to this headline">¶</a></h1>
<p>The code is from the HOMER documentation:</p>
<p><a class="reference external" href="http://homer.ucsd.edu/homer/ngs/csRNAseq/index.html">HOMER: csRNA-seq Analysis Tutorial</a></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
export PATH=/data/reddylab/software/homer/bin/:$PATH
makeTagDirectory
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stderr highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Usage: makeTagDirectory &lt;directory&gt; &lt;alignment file 1&gt; [file 2] ... [options]

	Creates a platform-independent &#39;tag directory&#39; for later analysis.
	Currently BED, eland, bowtie, and sam files are accepted. The program will try to
	automatically detect the alignment format if not specified.  Program will also
	unzip *.gz, *.bz2, and *.zip files and convert *.bam to sam files on the fly
	Existing tag directories can be added or combined to make a new one using -d/-t
	If more than one format is needed and the program cannot auto-detect it properly,
	make separate tag directories by running the program separately, then combine them.
	To perform QC/manipulations on an existing tag directory, add &quot;-update&quot;

	Options:
		-fragLength &lt;# | given | pe&gt; (Set estimated fragment length or use PE length - given: use read lengths)
			By default treats the sample as a single read ChIP-Seq experiment
		-format &lt;X&gt; where X can be: (with column specifications underneath)
			bed - BED format files:
				(1:chr,2:start,3:end,4:+/- or read name,5:# tags,6:+/-)
				-force5th (5th column of BED file contains # of reads mapping to position)
			sam - SAM formatted files (use samTools to covert BAMs into SAM if you have BAM)
				-unique (keep if there is a single best alignment based on mapq)
					-mapq &lt;#&gt; (Minimum mapq for -unique, default: 10, set negative to use AS:i:/XS:i:)
				-keepOne (keep one of the best alignments even if others exist)
				-keepAll (include all alignments in SAM file)
				-mis (Maximum allowed mismatches, default: no limit, uses MD:Z: tag)
				-sspe (strand specific, paired-end reads[flips strand of 2nd read to match])
				-read1/-read2 (only analyze 1st or 2nd read for PE sequencing)
			bowtie - output from bowtie (run with --best -k 2 options)
				(1:read name,2:+/-,3:chr,4:position,5:seq,6:quality,7:NA,8:misInfo)
			eland_result - output from basic eland
				(1:read name,2:seq,3:code,4:#zeroMM,5:#oneMM,6:#twoMM,7:chr,
							8:position,9:F/R,10-:mismatches
			eland_export - output from illumina pipeline (22 columns total)
				(1-5:read name info,9:sequence,10:quality,11:chr,13:position,14:strand)
			eland_extended - output from illumina pipeline (4 columns total)
				(1:read name,2:sequence,3:match stats,4:positions[,])
			mCpGbed - encode style mCpG reporting in extended BED format, no auto-detect
				(1:chr,2:start,3:end,4:name,5:,6:+/-,7:,8:,9:,10:#C,11:#mC)
			allC - Lister style output files detailing the read information about all cytosines
				(1:chr,2:pos,3:strand,4:context,#mC,#totalC,#unmC
			bismark - Bismark style output files detailing the read information about all cytosines
				(1:chr,2:pos,3:strand,4:#mC,5:#unmC,6:context,7:triseq
				-minCounts &lt;#&gt; (minimum number of reads to report mC/C ratios, default: 10)
				-mCcontext &lt;CG|CHG|CHH|all&gt; (only use C&#39;s in this context, default: CG)
			HiCsummary - minimal paired-end read mapping information
				(1:readname,2:chr1,3:5&#39;pos1,4:strand1,5:chr2,6:5&#39;pos2,7:strand2)
		-flip (flip strand of each read, i.e. might want to use with some RNA-seq)
		-totalReads &lt;#|all|default&gt; (set the effective total number of reads - all includes multimappers)
		-force5th (5th column of BED file contains # of reads mapping to position)
		-d &lt;tag directory&gt; [tag directory 2] ... (add Tag directory to new tag directory)
		-t &lt;tag file&gt; [tag file 2] ... (add tag file i.e. *.tags.tsv to new tag directory)
		-single (Create a single tags.tsv file for all &quot;chromosomes&quot; - i.e. if &gt;100 chromosomes)
		-update (Use current tag directory for QC/processing, do not parse new alignment files)
		-tbp &lt;#&gt; (Maximum tags per bp, default: no maximum)
		-precision &lt;1|2|3&gt; (number of decimal places to use for tag totals, default: 1)
		-minlen &lt;#&gt; and -maxlen &lt;#&gt; (Filter reads with lengths outside this range)

		GC-bias options:
		-genome &lt;genome version&gt; (To see available genomes, use &quot;-genome list&quot;)
			-or- (for custom genomes):
		-genome &lt;path-to-FASTA file or directory of FASTA files&gt;

		-checkGC (check Sequence bias, requires &quot;-genome&quot;)
			-freqStart &lt;#&gt; (offset to start calculating frequency, default: -50)
			-freqEnd &lt;#&gt; (distance past fragment length to calculate frequency, default: +50)
			-oligoStart &lt;#&gt; (oligo bias start)
			-oligoEnd &lt;#&gt; (oligo bias end)
		-normGC &lt;target GC profile file&gt; (i.e. tagGCcontent.txt file from control experiment)
			Use &quot;-normGC default&quot; to match the genomic GC distribution
		-normFixedOligo &lt;oligoFreqFile&gt; (normalize 5&#39; end bias, &quot;-normFixedOligo default&quot; ok)
		-minNormRatio &lt;#&gt; (Minimum deflation ratio of tag counts, default: 0.25)
		-maxNormRatio &lt;#&gt; (Maximum inflation ratio of tag counts, default: 2.0)
		-iterNorm &lt;#&gt; (Sets -max/minNormRatio to 1 and 0, iteratively normalizes such that the
			resulting distrubtion is no more than #% different than target, i.e. 0.1,default: off)
		-filterReads &lt;seq&gt; &lt;offset&gt; &lt;keep|remove&gt; (filter reads based on oligo sequence in the genome)

	HiC options
		-removePEbg (remove paired end tags within 1.5x fragment length on same chr)
			-PEbgLength &lt;#&gt; (remove PE  reads facing on another within this distance, default: 1.5x fragLen)
		-restrictionSite &lt;seq&gt; (i.e. AAGCTT for HindIII, assign data &lt; 1.5x fragment length to sites)
			Must specify genome sequence directory too. (-rsmis &lt;#&gt; to specify mismatches, def: 0)
			-both, -one, -onlyOne, -none (Keeps reads near restriction sites, default: keep all)
			-removeSelfLigation (removes reads linking same restriction fragment)
			-removeRestrictionEnds (removes reads starting on a restriction fragment)
			-assignMidPoint (will place reads in the middle of HindIII fragments)
			-restrictionSiteLength &lt;#&gt; (maximum distance from restriction site, default: 1.5x fragLen)
		-removeSpikes &lt;size bp&gt; &lt;#&gt; (remove tags from regions with &gt; than # times
			the average tags per size bp, suggest &quot;-removeSpikes 10000 8&quot;)
		-bowtiePE (PE alignments in bowtie alignment, assumes last character of read name is 0 or 1)
			(don&#39;t need this for sam/bam files)
</pre></div>
</div>
</div>
</div>
<div class="section" id="homer-maketagdirectory-of-single-sample-option-genome-hg38">
<h2><span class="section-number">2.2.1. </span>HOMER makeTagDirectory of single sample (Option: genome hg38)<a class="headerlink" href="#homer-maketagdirectory-of-single-sample-option-genome-hg38" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/maketag_t00_hg38.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_CAP=/data/reddylab/Kuei/Dex_ProCap
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads

### PRO-cap
FP_BAM_CAP_T00=$FD_CAP/new_files/A549_control_alignments/A549_untreated_merged.bam
FP_BAM_CAP_T15=$FD_CAP/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
FP_BAM_CAP_T60=$FD_CAP/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam

### make Tag dir for PRO-cap (control)
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_ProCap/run_homer
FD_TAG=$FD_OUT/tags/procap_t00_hg38
FP_BAM=$FP_BAM_CAP_T00
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24611910
</pre></div>
</div>
</div>
</div>
<div class="section" id="quality-control-checks">
<h3><span class="section-number">2.2.1.1. </span>Quality Control Checks<a class="headerlink" href="#quality-control-checks" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span>  <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">FD_TAG</span><span class="o">=</span><span class="s2">&quot;/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00_hg38&quot;</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagCountDistribution.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tags per tag position (Median = 0, tags per genomic bp = 0.003)</th>
      <th>Fraction of Positions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.845399</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.125846</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.016180</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3</td>
      <td>0.005331</td>
    </tr>
    <tr>
      <th>4</th>
      <td>4</td>
      <td>0.002493</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Tags per tag position (Median = 0, tags per genomic bp = 0.003)&#39;</span><span class="p">]</span>
<span class="n">y</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Fraction of Positions&#39;</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Tag Position&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Fraction of Positions&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_9_0.png" src="../../_images/02_homer_tag_9_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagLengthDistribution.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tag Length (Average tag length = 61.291551)</th>
      <th>Fraction of Tags</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.000000</td>
    </tr>
    <tr>
      <th>3</th>
      <td>3</td>
      <td>0.000118</td>
    </tr>
    <tr>
      <th>4</th>
      <td>4</td>
      <td>0.000088</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Tag Length (Average tag length = 61.291551)&#39;</span><span class="p">]</span>
<span class="n">y</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Fraction of Tags&#39;</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Tag Length&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Fraction of Tags&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_11_0.png" src="../../_images/02_homer_tag_11_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagFreqUniq.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Offset</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>A/T</th>
      <th>C/G</th>
      <th>A/G</th>
      <th>C/T</th>
      <th>AA</th>
      <th>...</th>
      <th>GA</th>
      <th>GC</th>
      <th>GG</th>
      <th>GT</th>
      <th>TA</th>
      <th>TC</th>
      <th>TG</th>
      <th>TT</th>
      <th>N</th>
      <th>NN</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>-50</td>
      <td>0.195888</td>
      <td>0.305542</td>
      <td>0.305853</td>
      <td>0.192717</td>
      <td>0.157873</td>
      <td>0.381803</td>
      <td>0.281499</td>
      <td>0.277863</td>
      <td>0.049858</td>
      <td>...</td>
      <td>0.057133</td>
      <td>0.098845</td>
      <td>0.105627</td>
      <td>0.044249</td>
      <td>0.027578</td>
      <td>0.057619</td>
      <td>0.059657</td>
      <td>0.047864</td>
      <td>13797127.0</td>
      <td>13797105.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>-49</td>
      <td>0.195351</td>
      <td>0.305991</td>
      <td>0.306543</td>
      <td>0.192114</td>
      <td>0.157543</td>
      <td>0.382477</td>
      <td>0.281228</td>
      <td>0.277252</td>
      <td>0.049264</td>
      <td>...</td>
      <td>0.057143</td>
      <td>0.098949</td>
      <td>0.106061</td>
      <td>0.044391</td>
      <td>0.027606</td>
      <td>0.057014</td>
      <td>0.059738</td>
      <td>0.047758</td>
      <td>13797123.0</td>
      <td>13797104.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>-48</td>
      <td>0.194626</td>
      <td>0.304944</td>
      <td>0.307455</td>
      <td>0.192975</td>
      <td>0.157195</td>
      <td>0.382912</td>
      <td>0.281340</td>
      <td>0.277557</td>
      <td>0.049341</td>
      <td>...</td>
      <td>0.056985</td>
      <td>0.099547</td>
      <td>0.106359</td>
      <td>0.044564</td>
      <td>0.027631</td>
      <td>0.057833</td>
      <td>0.059695</td>
      <td>0.047816</td>
      <td>13797127.0</td>
      <td>13797112.0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>-47</td>
      <td>0.194433</td>
      <td>0.306048</td>
      <td>0.307269</td>
      <td>0.192250</td>
      <td>0.156566</td>
      <td>0.383610</td>
      <td>0.280810</td>
      <td>0.277565</td>
      <td>0.049180</td>
      <td>...</td>
      <td>0.056959</td>
      <td>0.099704</td>
      <td>0.105918</td>
      <td>0.044689</td>
      <td>0.027638</td>
      <td>0.057645</td>
      <td>0.059460</td>
      <td>0.047508</td>
      <td>13797131.0</td>
      <td>13797112.0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>-46</td>
      <td>0.194335</td>
      <td>0.306519</td>
      <td>0.307208</td>
      <td>0.191938</td>
      <td>0.156417</td>
      <td>0.383667</td>
      <td>0.280535</td>
      <td>0.277823</td>
      <td>0.049091</td>
      <td>...</td>
      <td>0.056934</td>
      <td>0.099563</td>
      <td>0.105990</td>
      <td>0.044721</td>
      <td>0.027432</td>
      <td>0.057406</td>
      <td>0.059480</td>
      <td>0.047620</td>
      <td>13797126.0</td>
      <td>13797111.0</td>
    </tr>
  </tbody>
</table>
<p>5 rows × 27 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="mi">5</span><span class="p">]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Offset&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_13_0.png" src="../../_images/02_homer_tag_13_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagAutocorrelation.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Distance in bp(Fragment Length Estimate: 223)(Peak Width Estimate: 223)</th>
      <th>Same Strand (+ for Watson strand, - for Crick)</th>
      <th>Opposite Strand</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>-2000</td>
      <td>14384.0</td>
      <td>13630.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>-1999</td>
      <td>14093.0</td>
      <td>13521.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>-1998</td>
      <td>14240.0</td>
      <td>13653.0</td>
    </tr>
    <tr>
      <th>3</th>
      <td>-1997</td>
      <td>14321.0</td>
      <td>13585.0</td>
    </tr>
    <tr>
      <th>4</th>
      <td>-1996</td>
      <td>14595.0</td>
      <td>13696.0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Distance in bp(Fragment Length Estimate: 223)(Peak Width Estimate: 223)&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Distance in bp&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Read counts relative to each 5&#39; end&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">600</span><span class="p">,</span> <span class="mi">600</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_15_0.png" src="../../_images/02_homer_tag_15_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Distance in bp(Fragment Length Estimate: 223)(Peak Width Estimate: 223)&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Distance in bp&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Read counts relative to each 5&#39; end&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">150</span><span class="p">,</span> <span class="mi">150</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_16_0.png" src="../../_images/02_homer_tag_16_0.png" />
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="homer-maketagdirectory-of-single-sample-option-maxlen-65">
<h2><span class="section-number">2.2.2. </span>HOMER makeTagDirectory of single sample (Option: maxlen 65)<a class="headerlink" href="#homer-maketagdirectory-of-single-sample-option-maxlen-65" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_ProCap/run_homer/log/maketag_t00_maxlen.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_CAP=/data/reddylab/Kuei/Dex_ProCap
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads

### PRO-cap
FP_BAM_CAP_T00=$FD_CAP/new_files/A549_control_alignments/A549_untreated_merged.bam
FP_BAM_CAP_T15=$FD_CAP/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
FP_BAM_CAP_T60=$FD_CAP/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam

### make Tag dir for PRO-cap (control)
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_ProCap/run_homer
FD_TAG=$FD_OUT/tags/procap_t00_maxlen
FP_BAM=$FP_BAM_CAP_T00
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC -maxlen 65

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24691150
</pre></div>
</div>
</div>
</div>
<div class="section" id="id1">
<h3><span class="section-number">2.2.2.1. </span>Quality Control Checks:<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span>  <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">FD_TAG</span><span class="o">=</span><span class="s2">&quot;/gpfs/fs1/data/reddylab/Kuei/Dex_ProCap/run_homer/tags/procap_t00_maxlen/&quot;</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="read-clonality-pcr-duplicates">
<h4><span class="section-number">2.2.2.1.1. </span>Read clonality/PCR duplicates<a class="headerlink" href="#read-clonality-pcr-duplicates" title="Permalink to this headline">¶</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagCountDistribution.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tags per tag position (Median = 0, tags per genomic bp = 0.002)</th>
      <th>Fraction of Positions</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.857530</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.114756</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.015106</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span><span class="o">=</span><span class="n">dat</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">dat</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;Tags per tag position&#39;</span><span class="p">)]</span>
<span class="n">y</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Fraction of Positions&#39;</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Tag Position&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Fraction of Positions&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_24_0.png" src="../../_images/02_homer_tag_24_0.png" />
</div>
</div>
</div>
<div class="section" id="read-length-distribution">
<h4><span class="section-number">2.2.2.1.2. </span>Read length distribution<a class="headerlink" href="#read-length-distribution" title="Permalink to this headline">¶</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagLengthDistribution.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tag Length (Average tag length = 39.173910)</th>
      <th>Fraction of Tags</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">x</span><span class="o">=</span><span class="n">dat</span><span class="o">.</span><span class="n">loc</span><span class="p">[:,</span> <span class="n">dat</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s1">&#39;Tag Length&#39;</span><span class="p">)]</span>
<span class="n">y</span><span class="o">=</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;Fraction of Tags&#39;</span><span class="p">]</span>

<span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">x</span><span class="p">,</span> <span class="n">y</span><span class="p">,</span> <span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="n">markersize</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Tag Length&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Fraction of Tags&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylim</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mf">0.05</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_27_0.png" src="../../_images/02_homer_tag_27_0.png" />
</div>
</div>
</div>
<div class="section" id="nucleotide-preferences-near-the-5-end-of-the-read">
<h4><span class="section-number">2.2.2.1.3. </span>Nucleotide preferences near the 5’ end of the read<a class="headerlink" href="#nucleotide-preferences-near-the-5-end-of-the-read" title="Permalink to this headline">¶</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagFreqUniq.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Offset</th>
      <th>A</th>
      <th>C</th>
      <th>G</th>
      <th>T</th>
      <th>A/T</th>
      <th>C/G</th>
      <th>A/G</th>
      <th>C/T</th>
      <th>AA</th>
      <th>...</th>
      <th>GA</th>
      <th>GC</th>
      <th>GG</th>
      <th>GT</th>
      <th>TA</th>
      <th>TC</th>
      <th>TG</th>
      <th>TT</th>
      <th>N</th>
      <th>NN</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>-50</td>
      <td>0.195517</td>
      <td>0.306309</td>
      <td>0.307169</td>
      <td>0.191005</td>
      <td>0.155278</td>
      <td>0.383424</td>
      <td>0.282875</td>
      <td>0.277222</td>
      <td>0.049888</td>
      <td>...</td>
      <td>0.057578</td>
      <td>0.099545</td>
      <td>0.106105</td>
      <td>0.043942</td>
      <td>0.026739</td>
      <td>0.057591</td>
      <td>0.059744</td>
      <td>0.046931</td>
      <td>8216937.0</td>
      <td>8216932.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>-49</td>
      <td>0.195189</td>
      <td>0.306890</td>
      <td>0.307778</td>
      <td>0.190144</td>
      <td>0.154816</td>
      <td>0.384271</td>
      <td>0.282808</td>
      <td>0.276575</td>
      <td>0.049275</td>
      <td>...</td>
      <td>0.057518</td>
      <td>0.099660</td>
      <td>0.106645</td>
      <td>0.043956</td>
      <td>0.026672</td>
      <td>0.056908</td>
      <td>0.059781</td>
      <td>0.046783</td>
      <td>8216935.0</td>
      <td>8216931.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>-48</td>
      <td>0.194421</td>
      <td>0.305943</td>
      <td>0.308844</td>
      <td>0.190791</td>
      <td>0.154572</td>
      <td>0.384725</td>
      <td>0.282785</td>
      <td>0.276677</td>
      <td>0.049341</td>
      <td>...</td>
      <td>0.057394</td>
      <td>0.100218</td>
      <td>0.106930</td>
      <td>0.044303</td>
      <td>0.026703</td>
      <td>0.057552</td>
      <td>0.059658</td>
      <td>0.046878</td>
      <td>8216937.0</td>
      <td>8216933.0</td>
    </tr>
  </tbody>
</table>
<p>3 rows × 27 columns</p>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">dat</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="mi">5</span><span class="p">]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Offset&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_30_0.png" src="../../_images/02_homer_tag_30_0.png" />
</div>
</div>
</div>
<div class="section" id="distribution-of-reads-relative-to-one-another">
<h4><span class="section-number">2.2.2.1.4. </span>Distribution of reads relative to one another<a class="headerlink" href="#distribution-of-reads-relative-to-one-another" title="Permalink to this headline">¶</a></h4>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fdiry</span> <span class="o">=</span> <span class="n">FD_TAG</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagAutocorrelation.txt&quot;</span>
<span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>

<span class="n">dat</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">dat</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Distance in bp(Fragment Length Estimate: 224)(Peak Width Estimate: 224)</th>
      <th>Same Strand (+ for Watson strand, - for Crick)</th>
      <th>Opposite Strand</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>-2000</td>
      <td>11636.0</td>
      <td>11299.0</td>
    </tr>
    <tr>
      <th>1</th>
      <td>-1999</td>
      <td>11608.0</td>
      <td>11571.0</td>
    </tr>
    <tr>
      <th>2</th>
      <td>-1998</td>
      <td>11355.0</td>
      <td>11922.0</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">val</span> <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">columns</span><span class="o">.</span><span class="n">values</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">str</span><span class="p">)</span>
<span class="n">idx</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">char</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="n">val</span><span class="p">,</span> <span class="s2">&quot;Distance&quot;</span><span class="p">)</span>

<span class="n">dat</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">val</span><span class="p">[</span><span class="n">idx</span><span class="p">][</span><span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">()</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlabel</span><span class="p">(</span><span class="s2">&quot;Distance in bp&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">ylabel</span><span class="p">(</span><span class="s2">&quot;Read counts relative to each 5&#39; end&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">150</span><span class="p">,</span> <span class="mi">150</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_33_0.png" src="../../_images/02_homer_tag_33_0.png" />
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="run-maketagdirectory-for-all-sample">
<h2><span class="section-number">2.2.3. </span>Run makeTagDirectory for all sample<a class="headerlink" href="#run-maketagdirectory-for-all-sample" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_CAP=/data/reddylab/Kuei/Dex_ProCap
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads

### PRO-cap
FP_BAM_CAP_T00=$FD_CAP/new_files/A549_control_alignments/A549_untreated_merged.bam
FP_BAM_CAP_T15=$FD_CAP/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
FP_BAM_CAP_T60=$FD_CAP/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam

### make Tag dir for PRO-cap (control)
FD_OUT=/gpfs/fs1/data/reddylab/Kuei/Dex_ProCap/run_homer
FD_TAG=$FD_OUT/tags/procap_t00_hg38
FP_BAM=$FP_BAM_CAP_T00
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC
</pre></div>
</div>
<p>Creating Tag Directories</p>
<p><code class="docutils literal notranslate"><span class="pre">makeTagDirectory</span> <span class="pre">&lt;Output</span> <span class="pre">Directory</span> <span class="pre">Name&gt;</span> <span class="pre">[options]</span> <span class="pre">&lt;alignment</span> <span class="pre">file1&gt;</span> <span class="pre">[alignment</span> <span class="pre">file</span> <span class="pre">2]</span> <span class="pre">...</span></code></p>
<div class="section" id="pro-cap-all-samples">
<h3><span class="section-number">2.2.3.1. </span>PRO-cap (all samples)<a class="headerlink" href="#pro-cap-all-samples" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold -Color-Bold-Blue">new_files</span>/  new_files.tar.gz  <span class=" -Color -Color-Bold -Color-Bold-Blue">out_new_files</span>/  <span class=" -Color -Color-Bold -Color-Bold-Blue">run_homer</span>/

</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="mi">1</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">run_homer</span><span class="o">/</span><span class="n">tags</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t00</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t00_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t15</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t15_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t60</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t60_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t00</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t15</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t60</span>/

</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="mi">1</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">new_files</span><span class="o">/*/*</span><span class="n">bam</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/data/reddylab/Kuei/Dex_PROcap/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
/data/reddylab/Kuei/Dex_PROcap/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged_with_dup.bam
/data/reddylab/Kuei/Dex_PROcap/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam
/data/reddylab/Kuei/Dex_PROcap/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged_with_dup.bam
/data/reddylab/Kuei/Dex_PROcap/new_files/A549_control_alignments/A549_untreated_merged.bam
/data/reddylab/Kuei/Dex_PROcap/new_files/A549_control_alignments/A549_untreated_merged_with_dup.bam
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/maketag_cap_all.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_CAP=/data/reddylab/Kuei/Dex_PROcap
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads

### PRO-cap
FP_BAM_CAP_T00=$FD_CAP/new_files/A549_control_alignments/A549_untreated_merged.bam
FP_BAM_CAP_T15=$FD_CAP/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
FP_BAM_CAP_T60=$FD_CAP/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam

### make Tag dir for PRO-cap (control)
FD_OUT=$FD_CAP/run_homer
FD_TAG=$FD_OUT/tags/procap_t00
FP_BAM=$FP_BAM_CAP_T00
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC

### make Tag dir for PRO-cap (15m)
FD_OUT=$FD_CAP/run_homer
FD_TAG=$FD_OUT/tags/procap_t15
FP_BAM=$FP_BAM_CAP_T15
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC

### make Tag dir for PRO-cap (1hr)
FD_OUT=$FD_CAP/run_homer
FD_TAG=$FD_OUT/tags/procap_t60
FP_BAM=$FP_BAM_CAP_T60
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24719864
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="pro-cap-all-samples-options-maxlen">
<h3><span class="section-number">2.2.3.2. </span>PRO-cap (all samples; options: maxlen)<a class="headerlink" href="#pro-cap-all-samples-options-maxlen" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/maketag_cap_all_maxlen.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads
FD_CAP=/data/reddylab/Kuei/Dex_PROcap
FD_OUT=/data/reddylab/Kuei/Dex_PROcap/run_homer

### PRO-cap
FP_BAM_CAP_T00=$FD_CAP/new_files/A549_control_alignments/A549_untreated_merged.bam
FP_BAM_CAP_T15=$FD_CAP/new_files/A549_15min_Dexamethasone_alignments/A549_15min_Dexamethasone_merged.bam
FP_BAM_CAP_T60=$FD_CAP/new_files/A549_1hr_Dexamethasone_alignments/A549_1h_Dexamethasone_merged.bam

### make Tag dir for PRO-cap (control)
FD_TAG=$FD_OUT/tags/procap_t00_maxlen
FP_BAM=$FP_BAM_CAP_T00
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC -maxlen 65

### make Tag dir for PRO-cap (15m)
FD_TAG=$FD_OUT/tags/procap_t15_maxlen
FP_BAM=$FP_BAM_CAP_T15
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC -maxlen 65

### make Tag dir for PRO-cap (1hr)
FD_TAG=$FD_OUT/tags/procap_t60_maxlen
FP_BAM=$FP_BAM_CAP_T60
makeTagDirectory $FD_TAG/ $FP_BAM -genome hg38 -checkGC -maxlen 65

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24719871
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="rna-seq-3-time-points">
<h3><span class="section-number">2.2.3.3. </span>RNA-seq (3 time points)<a class="headerlink" href="#rna-seq-3-time-points" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">projects</span><span class="o">/</span><span class="n">GGR</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">rna_seq</span><span class="o">/</span><span class="n">mapped_reads</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold -Color-Bold-Blue">dex_depletion</span>/               <span class=" -Color -Color-Bold -Color-Bold-Blue">iter-1</span>/                       <span class=" -Color -Color-Bold -Color-Bold-Blue">iter_dex_depletion</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">iter0</span>/                       <span class=" -Color -Color-Bold -Color-Bold-Blue">iter-1_cell_line_comparison</span>/  <span class=" -Color -Color-Bold -Color-Bold-Blue">iter_hCas9_puro</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">iter0_cell_line_comparison</span>/  <span class=" -Color -Color-Bold -Color-Bold-Blue">iter-1_HDAC_inhibitors</span>/       <span class=" -Color -Color-Bold -Color-Bold-Blue">iter_over</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">iter0_HDAC_inhibitors</span>/       <span class=" -Color -Color-Bold -Color-Bold-Blue">iter_dCas9_krab_puro</span>/         <span class=" -Color -Color-Bold -Color-Bold-Blue">iter_short</span>/

</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>ls /data/reddylab/projects/GGR/data/rna_seq/mapped_reads/iter0/t00_rep?/
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>/data/reddylab/projects/GGR/data/rna_seq/mapped_reads/iter0/t00_rep1/:
STAR_1pass  STAR_2pass_featurecounts  STAR_2pass_rsem

/data/reddylab/projects/GGR/data/rna_seq/mapped_reads/iter0/t00_rep2/:
STAR_1pass  STAR_2pass_featurecounts  STAR_2pass_rsem

/data/reddylab/projects/GGR/data/rna_seq/mapped_reads/iter0/t00_rep3/:
STAR_1pass  STAR_2pass_featurecounts  STAR_2pass_rsem

/data/reddylab/projects/GGR/data/rna_seq/mapped_reads/iter0/t00_rep4/:
STAR_1pass  STAR_2pass_featurecounts  STAR_2pass_rsem
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">%%</span><span class="k">bash</span>
module load perl
module load gcc
source /data/reddylab/software/miniconda2/bin/activate alex_dev
export PATH=/data/reddylab/software/homer/bin/:$PATH
sbatch -pnew,all \
    --mem 16G \
    -o /gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/log/maketag_rna_all.txt \
    &lt;&lt;&#39;EOF&#39;
#!/bin/bash

### set directory
FP_GEN=/data/reddylab/Kuei/annotation/gencode.v34.annotation.gtf
FD_GGR=/data/reddylab/projects/GGR/
FD_RNA_ALIGN=$FD_GGR/data/rna_seq/mapped_reads
FD_CAP=/data/reddylab/Kuei/Dex_PROcap
FD_OUT=/data/reddylab/Kuei/Dex_PROcap/run_homer

### RNA-seq
FP_BAM_RNA_T00_short_rep1=$FD_RNA_ALIGN/iter_short/t00m_rep1.star2.Aligned.out.sorted.bam
FP_BAM_RNA_T00_short_rep2=$FD_RNA_ALIGN/iter_short/t00m_rep2.star2.Aligned.out.sorted.bam
FP_BAM_RNA_T00_short_rep3=$FD_RNA_ALIGN/iter_short/t00m_rep3.star2.Aligned.out.sorted.bam

FP_BAM_RNA_T00_long_rep1=$FD_RNA_ALIGN/iter0/t00_rep1/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T00_long_rep2=$FD_RNA_ALIGN/iter0/t00_rep2/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T00_long_rep3=$FD_RNA_ALIGN/iter0/t00_rep3/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T00_long_rep4=$FD_RNA_ALIGN/iter0/t00_rep4/STAR_2pass_featurecounts/Aligned.out.sorted.bam

FP_BAM_RNA_T15_short_rep1=$FD_RNA_ALIGN/iter_short/t15m_rep1.star2.Aligned.out.sorted.bam
FP_BAM_RNA_T15_short_rep2=$FD_RNA_ALIGN/iter_short/t15m_rep2.star2.Aligned.out.sorted.bam
FP_BAM_RNA_T15_short_rep3=$FD_RNA_ALIGN/iter_short/t15m_rep3.star2.Aligned.out.sorted.bam

FP_BAM_RNA_T60_long_rep1=$FD_RNA_ALIGN/iter0/t1_rep1/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T60_long_rep2=$FD_RNA_ALIGN/iter0/t1_rep2/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T60_long_rep3=$FD_RNA_ALIGN/iter0/t1_rep3/STAR_2pass_featurecounts/Aligned.out.sorted.bam
FP_BAM_RNA_T60_long_rep4=$FD_RNA_ALIGN/iter0/t1_rep4/STAR_2pass_featurecounts/Aligned.out.sorted.bam

### make Tag dir for RNA-seq (control)
FD_TAG=$FD_OUT/tags/rnaseq_t00
FP_BAM1=$FP_BAM_RNA_T00_short_rep1
FP_BAM2=$FP_BAM_RNA_T00_short_rep2
FP_BAM3=$FP_BAM_RNA_T00_short_rep3
makeTagDirectory $FD_TAG/ $FP_BAM1 $FP_BAM2 $FP_BAM3 -genome hg38 -checkGC

### make Tag dir for RNA-seq (15m)
FD_TAG=$FD_OUT/tags/rnaseq_t15
FP_BAM1=$FP_BAM_RNA_T15_short_rep1
FP_BAM2=$FP_BAM_RNA_T15_short_rep2
FP_BAM3=$FP_BAM_RNA_T15_short_rep3
makeTagDirectory $FD_TAG/ $FP_BAM1 $FP_BAM2 $FP_BAM3 -genome hg38 -checkGC

### make Tag dir for RNA-seq (1hr)
FD_TAG=$FD_OUT/tags/rnaseq_t60
FP_BAM1=$FP_BAM_RNA_T60_long_rep1
FP_BAM2=$FP_BAM_RNA_T60_long_rep2
FP_BAM3=$FP_BAM_RNA_T60_long_rep3
FP_BAM4=$FP_BAM_RNA_T60_long_rep4
makeTagDirectory $FD_TAG/ $FP_BAM1 $FP_BAM2 $FP_BAM3 $FP_BAM4 -genome hg38 -checkGC

EOF
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>Submitted batch job 24719877
</pre></div>
</div>
</div>
</div>
<hr class="docutils" />
<p>makeTagDirectory basically parses through the alignment file and splits the tags into separate files based on their chromosome.  As a result, several *.tags.tsv files are created in the output directory.  These are made to very efficiently return to the data during downstream analysis.  This also helps speed up the analysis of very large data sets without running out of memory.</p>
<p>In the end, your output directory will contain several *.tags.tsv files, as well as a file named “tagInfo.txt”.</p>
<p>This file contains information about your sequencing run, including the total number of tags considered.  This file is used by later programs to quickly reference information about the experiment, and can be manually modified to set certain parameters for analysis.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head /data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/tagInfo.txt
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>name	Unique Positions	Total Tags
genome=hg38	13991522	9384553.0
fragmentLengthEstimate=223		
peakSizeEstimate=223		
tagsPerBP=0.002906		
averageTagsPerPosition=0.230		
medianTagsPerPosition=0		
averageTagLength=61.292		
gsizeEstimate=3229495724		
averageFragmentGCcontent=0.620		
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="o">!</span>head /data/reddylab/Kuei/Dex_PROcap/run_homer/tags/procap_t00/chr1.tags.tsv
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	chr1	13325	0	1.0	144
	chr1	13658	1	0.5	112
	chr1	13690	1	0.5	144
	chr1	16077	0	0.5	27
	chr1	16103	1	0.5	27
	chr1	16463	0	0.5	144
	chr1	16609	1	0.5	129
	chr1	17443	0	0.5	75
	chr1	17462	0	0.5	39
	chr1	17500	1	0.5	39
</pre></div>
</div>
</div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="check-output-message">
<h2><span class="section-number">2.2.4. </span>Check output message<a class="headerlink" href="#check-output-message" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">ls</span> <span class="o">-</span><span class="mi">1</span> <span class="o">/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">fs1</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">run_homer</span><span class="o">/</span><span class="n">tags</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span><span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t00</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t00_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t15</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t15_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t60</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">procap_t60_maxlen</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t00</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t15</span>/
<span class=" -Color -Color-Bold -Color-Bold-Blue">rnaseq_t60</span>/

</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="o">/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">fs1</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">run_homer</span><span class="o">/</span><span class="n">log</span><span class="o">/</span><span class="n">maketag_cap_all</span><span class="o">.</span><span class="n">txt</span> <span class="o">|</span> <span class="n">grep</span> <span class="n">Avg</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Avg Fragment GC% = 61.96%
	Avg Expected GC% = 40.60%
	Avg Fragment GC% = 60.89%
	Avg Expected GC% = 40.60%
	Avg Fragment GC% = 59.90%
	Avg Expected GC% = 40.63%
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="o">/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">fs1</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">run_homer</span><span class="o">/</span><span class="n">log</span><span class="o">/</span><span class="n">maketag_cap_all_maxlen</span><span class="o">.</span><span class="n">txt</span> <span class="o">|</span> <span class="n">grep</span> <span class="n">Avg</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>	Avg Fragment GC% = 62.16%
	Avg Expected GC% = 40.61%
	Avg Fragment GC% = 61.30%
	Avg Expected GC% = 40.61%
	Avg Fragment GC% = 60.71%
	Avg Expected GC% = 40.63%
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">cat</span> <span class="o">/</span><span class="n">gpfs</span><span class="o">/</span><span class="n">fs1</span><span class="o">/</span><span class="n">data</span><span class="o">/</span><span class="n">reddylab</span><span class="o">/</span><span class="n">Kuei</span><span class="o">/</span><span class="n">Dex_PROcap</span><span class="o">/</span><span class="n">run_homer</span><span class="o">/</span><span class="n">log</span><span class="o">/</span><span class="n">maketag_rna_all</span><span class="o">.</span><span class="n">txt</span> <span class="o">|</span> <span class="n">grep</span> <span class="n">Avg</span>
</pre></div>
</div>
</div>
</div>
</div>
<div class="section" id="id2">
<h2><span class="section-number">2.2.5. </span>Quality Control Checks<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h2>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">import</span> <span class="nn">numpy</span>  <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
<span class="kn">import</span> <span class="nn">seaborn</span> <span class="k">as</span> <span class="nn">sns</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">FD_OUT</span><span class="o">=</span><span class="s2">&quot;/gpfs/fs1/data/reddylab/Kuei/Dex_PROcap/run_homer/tags&quot;</span>
<span class="n">os</span><span class="o">.</span><span class="n">listdir</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>[&#39;procap_t00&#39;,
 &#39;procap_t60_maxlen&#39;,
 &#39;procap_t15&#39;,
 &#39;procap_t15_maxlen&#39;,
 &#39;procap_t60&#39;,
 &#39;procap_t00_maxlen&#39;,
 &#39;rnaseq_t00&#39;,
 &#39;rnaseq_t15&#39;,
 &#39;rnaseq_t60&#39;]
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">FDIRYS_PROCAP</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;procap_t00&#39;</span><span class="p">,</span>
    <span class="s1">&#39;procap_t15&#39;</span><span class="p">,</span>
    <span class="s1">&#39;procap_t60&#39;</span><span class="p">]</span>

<span class="n">FDIRYS_PROCAP_MAXLEN</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;procap_t00_maxlen&#39;</span><span class="p">,</span>
    <span class="s1">&#39;procap_t15_maxlen&#39;</span><span class="p">,</span>
    <span class="s1">&#39;procap_t60_maxlen&#39;</span><span class="p">]</span>

<span class="n">FDIRYS_RNASEQ</span> <span class="o">=</span> <span class="p">[</span>
    <span class="s1">&#39;rnaseq_t00&#39;</span><span class="p">,</span>
    <span class="s1">&#39;rnaseq_t15&#39;</span><span class="p">,</span>
    <span class="s1">&#39;rnaseq_t60&#39;</span><span class="p">]</span>

<span class="n">FDIRYS</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">r_</span><span class="p">[</span><span class="n">FDIRYS_PROCAP</span><span class="p">,</span> <span class="n">FDIRYS_PROCAP_MAXLEN</span><span class="p">,</span> <span class="n">FDIRYS_RNASEQ</span><span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id3">
<h3><span class="section-number">2.2.5.1. </span>Read clonality/PCR duplicates<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">### set file and column names</span>
<span class="n">cname</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;Tags per tag position&quot;</span><span class="p">,</span> <span class="s2">&quot;Fraction of Positions&quot;</span><span class="p">]</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagCountDistribution.txt&quot;</span>

<span class="c1">### read in the results (procap)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_PROCAP</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_procap</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>

<span class="c1">### read in the results (procap maxlen)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_PROCAP_MAXLEN</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_procap_maxlen</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>

<span class="c1">### read in the results (rnaseq)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_RNASEQ</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_rnaseq</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>
    
<span class="c1">###</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_procap</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_procap_maxlen</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_rnaseq</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tags per tag position</th>
      <th>Fraction of Positions</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.845399</td>
      <td>procap_t00</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.125846</td>
      <td>procap_t00</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.016180</td>
      <td>procap_t00</td>
    </tr>
  </tbody>
</table>
</div></div><div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tags per tag position</th>
      <th>Fraction of Positions</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.857530</td>
      <td>procap_t00_maxlen</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.114756</td>
      <td>procap_t00_maxlen</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.015106</td>
      <td>procap_t00_maxlen</td>
    </tr>
  </tbody>
</table>
</div></div><div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tags per tag position</th>
      <th>Fraction of Positions</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.601156</td>
      <td>rnaseq_t00</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.205465</td>
      <td>rnaseq_t00</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.065564</td>
      <td>rnaseq_t00</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">ncols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">wspace</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_procap</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_procap_maxlen</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>

<span class="n">ax</span> <span class="o">=</span> <span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_rnaseq</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">ax</span><span class="o">.</span><span class="n">set_yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>

<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_62_0.png" src="../../_images/02_homer_tag_62_0.png" />
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">### </span>
<span class="n">cname</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;Tags per tag position&quot;</span><span class="p">,</span> <span class="s2">&quot;Fraction of Positions&quot;</span><span class="p">]</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagCountDistribution.txt&quot;</span>

<span class="c1">###</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">ax</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span>
    <span class="n">data</span> <span class="o">=</span> <span class="n">dat_merge</span><span class="p">,</span>
    <span class="n">ax</span>   <span class="o">=</span> <span class="n">ax</span><span class="p">,</span>
    <span class="n">x</span>    <span class="o">=</span> <span class="s2">&quot;Tags per tag position&quot;</span><span class="p">,</span> 
    <span class="n">y</span>    <span class="o">=</span> <span class="s2">&quot;Fraction of Positions&quot;</span><span class="p">,</span> 
    <span class="n">hue</span>  <span class="o">=</span> <span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">xscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">yscale</span><span class="p">(</span><span class="s1">&#39;log&#39;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_64_0.png" src="../../_images/02_homer_tag_64_0.png" />
</div>
</div>
</div>
<div class="section" id="id4">
<h3><span class="section-number">2.2.5.2. </span>Read length distribution<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1">### </span>
<span class="n">cname</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;Tag Length&quot;</span><span class="p">,</span> <span class="s2">&quot;Fraction of Tags&quot;</span><span class="p">]</span>
<span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagLengthDistribution.txt&quot;</span>

<span class="c1">### read in the results (procap)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_PROCAP</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_procap</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>

<span class="c1">### read in the results (procap maxlen)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_PROCAP_MAXLEN</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_procap_maxlen</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>

<span class="c1">### read in the results (rnaseq)</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS_RNASEQ</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">names</span><span class="o">=</span><span class="n">cname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
<span class="n">dat_merge_rnaseq</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">concat</span><span class="p">(</span><span class="n">lst</span><span class="p">)</span>
    
<span class="c1">###</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_procap</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_procap_maxlen</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
<span class="n">display</span><span class="p">(</span><span class="n">dat_merge_rnaseq</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">))</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tag Length</th>
      <th>Fraction of Tags</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>procap_t00</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.0</td>
      <td>procap_t00</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.0</td>
      <td>procap_t00</td>
    </tr>
  </tbody>
</table>
</div></div><div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tag Length</th>
      <th>Fraction of Tags</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>procap_t00_maxlen</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.0</td>
      <td>procap_t00_maxlen</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.0</td>
      <td>procap_t00_maxlen</td>
    </tr>
  </tbody>
</table>
</div></div><div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>Tag Length</th>
      <th>Fraction of Tags</th>
      <th>sample</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>0</td>
      <td>0.0</td>
      <td>rnaseq_t00</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>0.0</td>
      <td>rnaseq_t00</td>
    </tr>
    <tr>
      <th>2</th>
      <td>2</td>
      <td>0.0</td>
      <td>rnaseq_t00</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">ncols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">3</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">wspace</span><span class="o">=</span><span class="mf">0.3</span><span class="p">)</span>

<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_procap</span><span class="p">,</span>        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_procap_maxlen</span><span class="p">,</span> <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">sns</span><span class="o">.</span><span class="n">lineplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">dat_merge_rnaseq</span><span class="p">,</span>        <span class="n">ax</span><span class="o">=</span><span class="n">axes</span><span class="p">[</span><span class="mi">2</span><span class="p">],</span> <span class="n">x</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">y</span><span class="o">=</span><span class="n">cname</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span> <span class="n">hue</span><span class="o">=</span><span class="s2">&quot;sample&quot;</span><span class="p">)</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_67_0.png" src="../../_images/02_homer_tag_67_0.png" />
</div>
</div>
<p>According to the document, most csRNA-seq reads are between 20-55 nt in length. In the PRO-cap data I have, most reads are around 140bp</p>
</div>
<div class="section" id="nucleotide-preferences">
<h3><span class="section-number">2.2.5.3. </span>Nucleotide preferences<a class="headerlink" href="#nucleotide-preferences" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagFreqUniq.txt&quot;</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">ncols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">9</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">wspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>

<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">ravel</span><span class="p">()</span>
<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">dat</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">lst</span><span class="p">):</span>
    <span class="n">dat</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,:</span><span class="mi">5</span><span class="p">]</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">&quot;Offset&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_71_0.png" src="../../_images/02_homer_tag_71_0.png" />
</div>
</div>
<p>According to the document:  <code class="docutils literal notranslate"><span class="pre">Most</span> <span class="pre">csRNA-seq</span> <span class="pre">data</span> <span class="pre">should</span> <span class="pre">have</span> <span class="pre">a</span> <span class="pre">strong</span> <span class="pre">preference</span> <span class="pre">for</span> <span class="pre">Initiator-like</span> <span class="pre">sequences</span> <span class="pre">near</span> <span class="pre">the</span> <span class="pre">5'</span> <span class="pre">end</span> <span class="pre">(usually</span> <span class="pre">C-1A0).</span> <span class="pre">Most</span> <span class="pre">species</span> <span class="pre">also</span> <span class="pre">show</span> <span class="pre">evidence</span> <span class="pre">for</span> <span class="pre">a</span> <span class="pre">TATA</span> <span class="pre">box</span> <span class="pre">at</span> <span class="pre">approx.</span> <span class="pre">-30</span> <span class="pre">bp</span> <span class="pre">(e.g.</span> <span class="pre">spike</span> <span class="pre">in</span> <span class="pre">A/T</span> <span class="pre">content)</span></code></p>
</div>
<div class="section" id="id5">
<h3><span class="section-number">2.2.5.4. </span>Distribution of reads relative to one another<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fname</span> <span class="o">=</span> <span class="s2">&quot;tagAutocorrelation.txt&quot;</span>
<span class="n">lst</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">fdiry</span> <span class="ow">in</span> <span class="n">FDIRYS</span><span class="p">:</span>
    <span class="n">fpath</span> <span class="o">=</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">FD_OUT</span><span class="p">,</span> <span class="n">fdiry</span><span class="p">,</span> <span class="n">fname</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">fpath</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
    <span class="n">dat</span>   <span class="o">=</span> <span class="n">dat</span><span class="o">.</span><span class="n">assign</span><span class="p">(</span><span class="n">sample</span><span class="o">=</span><span class="k">lambda</span> <span class="n">x</span><span class="p">:</span> <span class="n">fdiry</span><span class="p">)</span>
    <span class="n">lst</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">dat</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">ncols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">9</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">wspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>

<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">ravel</span><span class="p">()</span>
<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">dat</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">lst</span><span class="p">):</span>
    <span class="n">dat</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">dat</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Distance in bp&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Read counts relative to each 5&#39; end&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">600</span><span class="p">,</span> <span class="mi">600</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]))</span>
    
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;center left&#39;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">))</span>    
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_75_0.png" src="../../_images/02_homer_tag_75_0.png" />
</div>
</div>
<p>According to the document: <code class="docutils literal notranslate"><span class="pre">In</span> <span class="pre">the</span> <span class="pre">case</span> <span class="pre">of</span> <span class="pre">transcription</span> <span class="pre">initiation</span> <span class="pre">data,</span> <span class="pre">reads</span> <span class="pre">from</span> <span class="pre">TSS</span> <span class="pre">will</span> <span class="pre">typically</span> <span class="pre">cluster</span> <span class="pre">on</span> <span class="pre">the</span> <span class="pre">same</span> <span class="pre">strand</span> <span class="pre">within</span> <span class="pre">50-100nt</span> <span class="pre">from</span> <span class="pre">each</span> <span class="pre">other.</span> <span class="pre">Due</span> <span class="pre">to</span> <span class="pre">the</span> <span class="pre">[normally]</span> <span class="pre">bidirectional</span> <span class="pre">nature</span> <span class="pre">of</span> <span class="pre">transcription</span> <span class="pre">from</span> <span class="pre">regulatory</span> <span class="pre">elements,</span> <span class="pre">reads</span> <span class="pre">will</span> <span class="pre">also</span> <span class="pre">typically</span> <span class="pre">appear</span> <span class="pre">upstream</span> <span class="pre">of</span> <span class="pre">TSS</span> <span class="pre">on</span> <span class="pre">the</span> <span class="pre">opposite</span> <span class="pre">strand</span> <span class="pre">(i.e.</span> <span class="pre">bidirectional</span> <span class="pre">transcription).</span></code></p>
<p>Below I have zoomed in further to check if the peak of the opposite strand of PRO-cap is within 50-100nt.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">fig</span><span class="p">,</span> <span class="n">axes</span> <span class="o">=</span> <span class="n">plt</span><span class="o">.</span><span class="n">subplots</span><span class="p">(</span><span class="n">ncols</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">nrows</span><span class="o">=</span><span class="mi">3</span><span class="p">,</span> <span class="n">figsize</span><span class="o">=</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="mi">9</span><span class="p">))</span>
<span class="n">plt</span><span class="o">.</span><span class="n">subplots_adjust</span><span class="p">(</span><span class="n">hspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">,</span> <span class="n">wspace</span><span class="o">=</span><span class="mf">0.5</span><span class="p">)</span>

<span class="n">axes</span> <span class="o">=</span> <span class="n">axes</span><span class="o">.</span><span class="n">ravel</span><span class="p">()</span>
<span class="k">for</span> <span class="n">ax</span><span class="p">,</span> <span class="n">dat</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">axes</span><span class="p">,</span> <span class="n">lst</span><span class="p">):</span>
    <span class="n">dat</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="n">dat</span><span class="o">.</span><span class="n">columns</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span><span class="o">.</span><span class="n">plot</span><span class="o">.</span><span class="n">line</span><span class="p">(</span><span class="n">ax</span><span class="o">=</span><span class="n">ax</span><span class="p">,</span> <span class="n">legend</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlabel</span><span class="p">(</span><span class="s2">&quot;Distance in bp&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_ylabel</span><span class="p">(</span><span class="s2">&quot;Read counts relative to each 5&#39; end&quot;</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_xlim</span><span class="p">(</span><span class="o">-</span><span class="mi">150</span><span class="p">,</span> <span class="mi">150</span><span class="p">)</span>
    <span class="n">ax</span><span class="o">.</span><span class="n">set_title</span><span class="p">(</span><span class="n">np</span><span class="o">.</span><span class="n">unique</span><span class="p">(</span><span class="n">dat</span><span class="p">[</span><span class="s1">&#39;sample&#39;</span><span class="p">]))</span>
    
<span class="n">plt</span><span class="o">.</span><span class="n">legend</span><span class="p">(</span><span class="n">loc</span><span class="o">=</span><span class="s1">&#39;center left&#39;</span><span class="p">,</span> <span class="n">bbox_to_anchor</span><span class="o">=</span><span class="p">(</span><span class="mf">1.0</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">))</span>    
<span class="n">plt</span><span class="o">.</span><span class="n">show</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<img alt="../../_images/02_homer_tag_78_0.png" src="../../_images/02_homer_tag_78_0.png" />
</div>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./notebooks/12_run_homer_procap"
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        
        <div class='prev-next-bottom'>
            
    <a class='left-prev' id="prev-link" href="01_prep_files.html" title="previous page"><span class="section-number">2.1. </span>Check the file paths for the data</a>
    <a class='right-next' id="next-link" href="03_homer_vis.html" title="next page"><span class="section-number">2.3. </span>Create Genome Browser Visualization Files</a>

        </div>
        
        </div>
    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The Jupyter Book Community<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="../../_static/js/index.d3f166471bb80abb5163.js"></script>


    
  </body>
</html>